---
title: "scPagwas"
date: "Last Updated: `r format(Sys.time(), '%d, %B, %Y at %H:%M')`"
output:
  md_document:
    variant: markdown_github
editor_options:
  chunk_output_type: console
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# scPagwas

**scPagwas**, a polygenic principal component-scoring method that links scRNA-seq data with large-scale GWAS summary statistics to prioritize genetics-modulated cells associated with complex diseases.
**scPagwas** is able to prioritize disease-associated individual cells by integrating the scRNA-seq data with polygenic signals from GWAS.


![Graphical abstract](./docs/reference/figures/workflow_20220222.png)

## Installation

You can install the released version of scPagwas from [github](https://github.com/dengchunyu/scPagwas) with:

``` r
devtools::install_github("dengchunyu/scPagwas")

```

## Example 

### 1.Preprogress and create internal storage

```{r message=FALSE,results = "hide",warning = FALSE}
 library(scPagwas)
 suppressMessages(library(Seurat))
 suppressWarnings(library(SOAR))
 suppressMessages(library("dplyr"))
 #Input pathway gene list, you can construct with youself.
 data(Genes_by_pathway_kegg)
 #gene annotation files.
 data(block_annotation)
 #LD data
 data(chrom_ld)

 #1.start to run the wrapper functions for preprogress.
 Pagwas<-scPagwas_main(Pagwas = NULL,
                     gwas_data =system.file("extdata", "GWAS_summ_example.txt", package = "scPagwas"),
                     add_eqtls="OnlyTSS",
                     block_annotation = block_annotation,
                     Single_data = system.file("extdata", "scRNAexample.rds", package = "scPagwas"),
                     FilterSingleCell=F,
                     Pathway_list=Genes_by_pathway_kegg,
                     chrom_ld = chrom_ld,
                     scPagwasSession="scPagwasSession")

```
 
### 2. Cell types functions  
 
```{r message=FALSE,results = "hide",warning = FALSE}
  Objects()
# 7th: link_pwpca_block function start!! 
  Pagwas <- link_pwpca_block(Pagwas,scPagwasSession="scPagwasSession")
#8th: Pagwas_perform_regression function start!!
  Pagwas <- Pagwas_perform_regression(Pagwas, 
                                      iters = 200,
                                      n.cores=1,
                                      scPagwasSession="scPagwasSession")
``` 
#### Visualize the celltypes results.

 1.barplot
```{r Bootstrap_P_Barplot, message=FALSE,warning = FALSE}
Objects()
Bootstrap_P_Barplot(Pagwas=Pagwas,
                    figurenames = NULL,
                    width = 5,
                    height = 7,
                    do_plot=T,
                    title = "Test scPagwas")
```

 2.Forestplot for estimate values
```{r Bootstrap_estimate_Plot,message=FALSE,warning = FALSE}
Objects()
Bootstrap_estimate_Plot(Pagwas=Pagwas,
                        figurenames = NULL,
                        width = 9,
                        height = 7,
                        do_plot=T)

```

#### pathway network

```{r pathway_contribution_network,message=FALSE,warning = FALSE}

suppressMessages(require("WGCNA"))
suppressMessages(require("patchwork"))
suppressMessages(require("tidygraph"))
suppressMessages(require("ggraph"))
suppressMessages(require("igraph"))
#check the objects
Objects()
Sys.setenv(R_LOCAL_CACHE="scPagwasSession")
plot_pathway_contribution_network(
                  mat_datExpr=pca_cell_df,
                  vec_pathwaycontribution=Pagwas$Pathway_block_heritability,
                  vec_pathways_highlight=names(sort(Pagwas$Pathway_block_heritability,decreasing = T)[1:5]),
                  n_max_pathways=20,
                  igraph_algorithm = "drl",
                  fontface_labels="bold.italic",
                  color_edge = "#9D9D9D",
                  fontSize_label_lg=4,
                  fontSize_legend_lg=4,
                  fontSize_legend_xlg=4,
                  edge_thickness = 1,
                  do_plot=T
                  
  )

```
#######


### 3.Single cell function 
scPagwas_main is a function wrapper other process codes.
as the parameters are the same as Pagwas_main,we can inherit the Pagwas reuslt for save time.

```{r message=FALSE,results = "hide",warning = FALSE}
  #check the objects
  #Objects()
  Pagwas <- link_scCell_pwpca_block(Pagwas,scPagwasSession="scPagwasSession")
  Pagwas <- scPagwas_perform_score(Pagwas,scPagwasSession="scPagwasSession")
  #check the objects in Pagwas,if you don't need rerun, remove some objects
```


#### Visualize the scPagwas_main results.
##### Visualize the scPagwas_score of single cell data in UMAP or TSNE plot. 

```{r scPagwas_Visualization,message=FALSE,warning = FALSE}
 require("RColorBrewer")
 require("Seurat")
 require("SeuratObject")
 require("ggsci")
 #check the objects
 Objects()
 scRNAexample<-readRDS(system.file("extdata", "scRNAexample.rds", package = "scPagwas"))
 scPagwas_Visualization(scPagwas_score = Pagwas$scPagwas_score,
                        Single_data = scRNAexample,
                        Reduction = TRUE,
                        assay = "SCT",
                        cellpercent = 0.1,
                        filename = NULL,
                        FigureType = "tsne",
                        width = 7,
                        height = 7,
                        lowColor = "#FFBC80", highColor = "#FC4F4F",
                        size = 1,
                        title = "scPagwas_score",
                        do_plot = T)
```

##### Plot the barplot of the proportion of positive Cells in celltypes

```{r bar_positie_nagtive,message=FALSE,,warning = FALSE,fig.height=6, fig.width=6}
library("RColorBrewer")
library("ggplot2")

scPagwas_score <- Pagwas$scPagwas_score[intersect(colnames(scRNAexample),names(Pagwas$scPagwas_score))]
scRNAexample$scPagwas_score <- scPagwas_score
thre <- sort(scRNAexample$scPagwas_score, decreasing = T)[ncol(scRNAexample) * 0.1]
scRNAexample$positiveCells<-rep(0,ncol(scRNAexample))
scRNAexample$positiveCells[scRNAexample$scPagwas_score>=thre]<-1

plot_bar_positie_nagtive(seurat_obj=scRNAexample,
                              var_ident="positiveCells",
                              var_group="anno",
                              vec_group_colors=NULL,
                              f_color=colorRampPalette(brewer.pal(n=10, name="RdYlBu")),
                              do_plot = T)
```

##### Plot the barplot of the proportion of celltypes in positive Cell 

```{r bar_positie_nagtive2,message=FALSE,warning = FALSE}
plot_bar_positie_nagtive(seurat_obj=scRNAexample,
                              var_ident="anno",
                              var_group="positiveCells",
                              vec_group_colors=c("#E8D0B3","#7EB5A6"),
                              do_plot = T)
```

##### Plot the top5 heritability correlation genes in celltypes

```{r vln_Corgenes,message=FALSE,fig.height=6, fig.width=8,warning = FALSE}
top5genes<-rownames(Pagwas$gene_heritability_correlation)[order(Pagwas$gene_heritability_correlation,decreasing = T)[1:5]]
plot_vln_Corgenes(seurat_obj=scRNAexample,
             assay="RNA", slot="data",
             var_group="anno",
             vec_features=top5genes,
             vec_group_colors= pal_d3(alpha =0.5)(10),
             do_plot = T
             )
```

#### Single cell regression

```{r message=FALSE,results = "hide"}
    Pagwas <- scPagwas_perform_regression(Pagwas,scPagwasSession="scPagwasSession")

```

#### Change the parameter of nfeatures 

```{r, include = FALSE}
 data(Genes_by_pathway_kegg)
 #gene annotation files.
 data(block_annotation)
 #LD data
 data(chrom_ld)
 
   Pagwas2<-scPagwas_main(Pagwas = NULL,
                     gwas_data = "E:/RPakage/scPagwas/inst/extdata/GWAS_summ_example.txt",
                     add_eqtls="OnlyTSS",
                     nfeatures = 5000,
                     block_annotation = block_annotation,
                     Single_data = "E:/RPakage/scPagwas/inst/extdata/scRNAexample.rds",
                     FilterSingleCell=TRUE,
                     Pathway_list=Genes_by_pathway_kegg,
                     chrom_ld = chrom_ld,
                     scPagwasSession="scPagwasSession2")
 # rm(chrom_ld)
  #Sys.setenv(R_LOCAL_CACHE="scPagwasSession2")
  Pagwas2 <- link_pwpca_block(Pagwas2,scPagwasSession="scPagwasSession2")
  Pagwas2 <- Pagwas_perform_regression(Pagwas2, scPagwasSession="scPagwasSession2",
                                       iters = 200,n.cores=1)
  Bootstrap_P_Barplot(Pagwas=Pagwas2,title = "5000 Test scPagwas")
  Pagwas2 <- link_scCell_pwpca_block(Pagwas2,scPagwasSession="scPagwasSession2")
  Pagwas2 <- scPagwas_perform_score(Pagwas2,scPagwasSession="scPagwasSession2")
  
  scPagwas_Visualization(scPagwas_score = Pagwas2$scPagwas_score,
                        Single_data = scRNAexample,
                        Reduction = TRUE,
                        assay = "SCT",
                        cellpercent = 0.1,
                        filename = "scPagwas2_testFigure",
                        FigureType = "tsne",
                        width = 7,
                        height = 7,
                        lowColor = "#FFBC80", highColor = "#FC4F4F",
                        size = 1,
                        title = "scPagwas_score_5000")
  
```

#### remove the Objects in session. 

```{r removeCathe}
# empty the cache
Objects()
Remove(Objects())

```
The workfile is ongoing...
