---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# scPagwas

<!-- badges: start -->
<!-- badges: end -->

scPagwas, a polygenic principal component-scoring method that links scRNA-seq data with large-scale GWAS summary statistics to prioritize genetics-modulated cells associated with complex diseases.
scPagwas is able to prioritize disease-associated individual cells by integrating the scRNA-seq data with polygenic signals from GWAS.

## Installation

You can install the released version of scPagwas from [github](https://github.com/dengchunyu/scPagwas) with:

``` r
devtools::install_github("dengchunyu/scPagwas")

```

## Example 

### 1.Preprogress and create internal storage

```{r Pagwas_main}
 library(Seurat)
 library(scPagwas)
 library(SOAR)

 ## basic example code
 #Input pathway gene list, you can construct with youself.
 data(Genes_by_pathway_kegg)
 #Input GWAS summry data(data.frame).
 data(GWAS_summ_example)
 #gene annotation files.
 data(gtf_df)
 #Single cell example data.
 data(scRNAexample)
 #LD data
 data(chrom_ld)
 ##create a session for internal storage
# Sys.setenv(R_LOCAL_CACHE="scPagwasSession")

 #1.start to run the wrapper functions for preprogress.
 Pagwas<-scPagwas_main(Pagwas = NULL,
                     gwas_data = GWAS_summ_example,
                     add_eqtls="OnlyTSS",
                     block_annotation = gtf_df,
                     Single_data = scRNAexample,
                     FilterSingleCell=TRUE,
                     Pathway_list=Genes_by_pathway_kegg,
                     chrom_ld = chrom_ld,
                     scPagwasSession="scPagwasSession")


```
 
### 2. Cell types functions  
 
```{r link_pwpca_block}
 
   message(paste(utils::timestamp(quiet = T), ' ******* 7th: link_pwpca_block function start!! ********',sep = ''))
  #timestart<-Sys.time()
  Pagwas <- link_pwpca_block(Pagwas)
  
  message(paste(utils::timestamp(quiet = T), ' ******* 8th: Pagwas_perform_regression function start!! ********',sep = ''))
  Pagwas <- Pagwas_perform_regression(Pagwas, iters = 200,n.cores=1)
  
``` 
#### Visualize the celltypes results.

 1.barplot
```{r Bootstrap_P_Barplot}
Bootstrap_P_Barplot(Pagwas=Pagwas,title = "Test scPagwas")
```

 2.Forestplot for estimate values
```{r Bootstrap_estimate_Plot}

Bootstrap_estimate_Plot(Pagwas=Pagwas,figurenames = NULL)

```

### 3.Single cell function 
```{r link_scCell_pwpca_block}
 #scPagwas_main is a function wrapper other process codes
 #as the parameters are the same as Pagwas_main,we can inherit the Pagwas reuslt for save time.
  message(paste(utils::timestamp(quiet = T), ' ******* 7th: link_scCell_pwpca_block function start!! ********',sep = ''))
  Pagwas <- link_scCell_pwpca_block(Pagwas)

  message(paste(utils::timestamp(quiet = T), ' ******* 8th: scPagwas_perform_score function start!! ********',sep = ''))
  Pagwas <- scPagwas_perform_score(Pagwas)
  
```


#### Visualize the scPagwas_main results.
```{r scPagwas_Visualization}
 scPagwas_Visualization(scPagwas_score = Pagwas$scPagwas_score,
                        Single_data = scRNAexample,
                        Reduction = TRUE,
                        assay = "SCT",
                        cellpercent = 0.1,
                        filename = "scPagwas_testFigure",
                        FigureType = "tsne",
                        width = 7,
                        height = 7,
                        lowColor = "#FFBC80", highColor = "#FC4F4F",
                        size = 1,
                        title = "scPagwas_score")

```
#### Single cell regression

```{r scPagwas_perform_regression}
    message(paste(utils::timestamp(quiet = T), ' ******* 9th:scPagwas_perform_regression function start!! ********',sep = ''))
    Pagwas <- scPagwas_perform_regression(Pagwas)

```


```{r removeCathe}
# empty the cache
#Objects()
Remove(Objects())

```
The workfile is ongoing...
