---
title: "Example1.Bmmc_monocytecount_singlecell&celltype_vignette"
date: "Last Updated: `r format(Sys.time(), '%d, %B, %Y at %H:%M')`"
output:
  html_document:
    theme: flatly
    highlight: zenburn
    fig_width: 7
    fig_height: 6
    fig_caption: true
    df_print: paged
vignette: >
  %\VignetteIndexEntry{Example1.Bmmc_monocytecount_singlecell&celltype_vignette}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: sentence
  chunk_output_type: inline
---

```{r, include = FALSE}
library(knitr)
knitr::opts_chunk$set(
 collapse = TRUE,
 comment = "#>",
  fig.path = "figures/vignette-",
 out.width = "60%"
)
```

# Example1. BMMC scRNA-seq data for monocyte count trait

We use BMMC scRNA-seq data and monocyte count trait to test scPagwas.
The processed data can be download from here: monocytecount_prune_gwas_data.txt <https://1drv.ms/t/s!As-aKqXDnDUHi9Ub-q1D50G2S-sfjA?e=KYqJPv>

BMMC scRNA-seq data can be obtained and treated in `Data_input_preproccess_for_scPagwas`.

## 1. Compute the singlecell and celltype result for monocytecount trait

```{r message=FALSE, eval = FALSE}
library(scPagwas)
Pagwas<-scPagwas_main(Pagwas =NULL,
                     gwas_data ="/share/pub/dengcy/GWAS_Multiomics/compare/bloodtraits/monocytecount_prune_gwas_data.txt",
                     Single_data ="/share/pub/dengcy/GWAS_Multiomics/singlecelldata/Seu_Hema_data.rds",
                     output.prefix="monocytecount_scPagwas",
                     output.dirs="monocytecount_bmmc",
                     seruat_return=T,
                     Pathway_list=Genes_by_pathway_kegg,
                     ncores=2,
                     assay="RNA",
                     block_annotation = block_annotation,
                     chrom_ld = chrom_ld)
print(Pagwas)

```




## 2.

1.barplot

```{r Bootstrap_P_Barplot, message=FALSE,warning = FALSE}

Bootstrap_P_Barplot(p_results=Pagwas_data@misc$bootstrap_results$bp_value[-1],
                    p_names=rownames(Pagwas_data@misc$bootstrap_results)[-1],
                    figurenames = NULL,
                    width = 5,
                    height = 7,
                    do_plot=T,
                    title = "Test scPagwas")

```

#### Visualize the scPagwas_main results.

##### Visualize the scPagwas_score of single cell data in UMAP or TSNE plot.

```{r scPagwas_Visualization,message=FALSE,warning = FALSE}
 require("RColorBrewer")
 require("Seurat")
 require("SeuratObject")
 require("ggsci")
 #check the objects

 DimPlot(Pagwas_data,group.by = "anno",pt.size=1,reduction="umap",label = T, repel=TRUE)+ 
 umap_theme()+ggtitle("Test")+labs(x="TSNE",y="")+theme(aspect.ratio=1)
 
 
 scPagwas_Visualization(Single_data=Pagwas_data,
                        p_thre = 0.05,
                        FigureType = "umap",
                        width = 7,
                        height = 7,
                        lowColor = "white", 
                        highColor = "red",
                        output.dirs="scPagwastest_output",
                        size = 0.5,
                        do_plot = T)
 
```

##### Plot the barplot of the proportion of positive Cells in celltypes

```{r bar_positie_nagtive,message=FALSE,warning = FALSE,fig.height=6, fig.width=6}
library("RColorBrewer")
library("ggplot2")

plot_bar_positie_nagtive(seurat_obj=Pagwas_data,
                              var_ident="positiveCells",
                              var_group="anno",
                              p_thre = 0.01,
                              vec_group_colors=NULL,
                              f_color=colorRampPalette(brewer.pal(n=10, name="RdYlBu")),
                              do_plot = T)
```

##### Plot the barplot of the proportion of celltypes in positive Cell

```{r bar_positie_nagtive2,message=FALSE,warning = FALSE}
plot_bar_positie_nagtive(seurat_obj=Pagwas_data,
                              var_ident="anno",
                              var_group="positiveCells",
                              vec_group_colors=c("#E8D0B3","#7EB5A6"),
                              do_plot = T)
```

##### Plot the top5 heritability correlation genes in celltypes

```{r vln_Corgenes,message=FALSE,fig.height=6, fig.width=8,warning = FALSE}
top5genes<-rownames(Pagwas_data@misc$gene_heritability_correlation)[order(Pagwas_data@misc$gene_heritability_correlation,decreasing = T)[1:5]]

plot_vln_Corgenes(seurat_obj=Pagwas_data,
             assay="RNA", slot="data",
             var_group="anno",
             vec_features=top5genes,
             vec_group_colors= pal_d3(alpha =0.5)(10),
             do_plot = T
             )

```

##### Plot the heritability correlated Pathways for each celltypes

```{r message=FALSE,results = "hide",warning = FALSE}
  library(tidyverse)
  library("rhdf5")
 library(ggplot2)
 library(grDevices)
 library(stats)
 library(FactoMineR)
 library(scales)
 library(reshape2)
 library(ggdendro)
 library(grImport2)
 library(gridExtra)
 library(grid)
 library(sisal)

 source(system.file("extdata", "plot_scpathway_contri_dot.R", package = "scPagwas"))

plot_scpathway_dot(Pagwas=Pagwas_data,
                   celltypes=unique(Idents(Pagwas_data))[1:5],
                             topn_path_celltype=5,
                             filter_p=0.05,
                             max_logp=15,
                             display_max_sizes=F,
                             size_var ="logrankPvalue" ,
                             col_var="proportion",
                             shape.scale = 8,
                             cols.use=c("lightgrey", "#E45826"),
                             dend_x_var = "logrankPvalue",
                             dist_method="euclidean",
                             hclust_method="ward.D",
                             do_plot = T,
                             figurenames = NULL,
                             width = 7,
                             height = 7)
                             
```

##### Plot the heritability correlated genes

```{r message=FALSE,results = "hide",warning = FALSE}

heritability_cor_scatterplot(gene_heri_cor=Pagwas_data@misc$gene_heritability_correlation,
                             topn_genes_label=10,
                             color_low="#035397",
                             color_high ="#F32424",
                             color_mid = "white",
                             text_size=2,
                             do_plot=T,
                             max.overlaps =20,
                             width = 7,
                             height = 7)


```
