<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Bmmc_monocytecount_singlecell_celltype_vignette • scPagwas</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Bmmc_monocytecount_singlecell_celltype_vignette">
<meta property="og:description" content="scPagwas">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-12345678-0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-12345678-0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scPagwas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.10.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Bmmc_monocytecount_singlecell_celltype_vignette.html">Bmmc_monocytecount_singlecell_celltype_vignette</a>
    </li>
    <li>
      <a href="../articles/Data_input_preproccess_for_scPagwas.html">Data_input_preproccess_for_scPagwas</a>
    </li>
    <li>
      <a href="../articles/Multi_TraitFiles_for_one_SingleData_running.html">Multi_TraitFiles_for_one_SingleData_running</a>
    </li>
    <li>
      <a href="../articles/Running_scPagwas_steps_by_SubFunctions.html">Running_scPagwas_steps_by_SubFunctions</a>
    </li>
    <li>
      <a href="../articles/Split_big_scRNAseqData_and_integrate_result.html">Split_big_scRNAseqData_and_integrate_result</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Bmmc_monocytecount_singlecell_celltype_vignette</h1>
            
            <h4 data-toc-skip class="date">Last Updated: 19, 七月, 2022 at 14:16</h4>
      
      
      <div class="hidden name"><code>Bmmc_monocytecount_singlecell_celltype_vignette.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="preprocessed-data">Preprocessed data<a class="anchor" aria-label="anchor" href="#preprocessed-data"></a>
</h2>
<p>We use BMMC scRNA-seq data and monocyte count trait to test scPagwas. The processed data can be download from here: monocytecount_prune_gwas_data.txt <a href="https://1drv.ms/t/s!As-aKqXDnDUHi6sx7Hqblj2Sgl7P8w?e=wGAip7" class="external-link uri">https://1drv.ms/t/s!As-aKqXDnDUHi6sx7Hqblj2Sgl7P8w?e=wGAip7</a></p>
<p>BMMC scRNA-seq data can be obtained and treated in <code>Data_input_preproccess_for_scPagwas</code>.</p>
</div>
<div class="section level2">
<h2 id="compute-the-singlecell-and-celltype-result-for-monocytecount-trait">1. Compute the singlecell and celltype result for monocytecount trait<a class="anchor" aria-label="anchor" href="#compute-the-singlecell-and-celltype-result-for-monocytecount-trait"></a>
</h2>
<p>If we running scPagwas in multi-core in Server environment, there may cause an error: <code>Error: Two levels of parallelism are used. See</code>?a ssert_cores`` add this code before call in R environment:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">OPENBLAS_NUM_THREADS=</span>1</span></code></pre></div>
<p>There is no need to run this code in window system.</p>
<p>In this example, we run the scPagwas for usual step, both running singlecell and celltype process.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scPagwas</span><span class="op">)</span>
<span class="va">Pagwas</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/scPagwas_main.html">scPagwas_main</a></span><span class="op">(</span>Pagwas <span class="op">=</span><span class="cn">NULL</span>,
                     gwas_data <span class="op">=</span><span class="st">"monocytecount_prune_gwas_data.txt"</span>,
                     Single_data <span class="op">=</span><span class="st">"Seu_Hema_data.rds"</span>,
                     output.prefix<span class="op">=</span><span class="st">"monocytecount_scPagwas"</span>,
                     output.dirs<span class="op">=</span><span class="st">"monocytecount_bmmc"</span>,
                     seruat_return<span class="op">=</span><span class="cn">T</span>,
                     Pathway_list<span class="op">=</span><span class="va">Genes_by_pathway_kegg</span>,
                     ncores<span class="op">=</span><span class="fl">5</span>,
                     assay<span class="op">=</span><span class="st">"RNA"</span>,
                     singlecell<span class="op">=</span><span class="cn">T</span>, 
                     celltype<span class="op">=</span><span class="cn">T</span>,
                     block_annotation <span class="op">=</span> <span class="va">block_annotation</span>,
                     chrom_ld <span class="op">=</span> <span class="va">chrom_ld</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">Pagwas</span>,file <span class="op">=</span> <span class="st">"monocytecount_bmmc_Pagwas1.10.0.RData"</span><span class="op">)</span></code></pre></div>
<p>Sometime, we need to remove the objects in cache folder:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu">SOAR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SOAR/man/Objects.html" class="external-link">Objects</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">{</span>
 <span class="fu">SOAR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SOAR/man/Remove.html" class="external-link">Remove</a></span><span class="op">(</span><span class="fu">SOAR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SOAR/man/Objects.html" class="external-link">Objects</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> 
<span class="op">}</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="result-interpretation">2. Result interpretation<a class="anchor" aria-label="anchor" href="#result-interpretation"></a>
</h2>
<p>There are two types of result, Seruat format return result and files output;</p>
<div class="section level3">
<h3 id="pagwas-return-result-for-seruat_returntrue">2.1 Pagwas : return result for seruat_return=TRUE<a class="anchor" aria-label="anchor" href="#pagwas-return-result-for-seruat_returntrue"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">print</span>(Pagwas)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>An object of class Seurat </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="dv">16488</span> features across <span class="dv">35582</span> samples within <span class="dv">3</span> assays </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>Active assay<span class="sc">:</span> <span class="fu">RNA</span> (<span class="dv">15860</span> features, <span class="dv">5000</span> variable features)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a> <span class="dv">2</span> other assays present<span class="sc">:</span> scPagwasPaHeritability, scPagwasPaPca</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a> <span class="dv">3</span> dimensional reductions calculated<span class="sc">:</span> pca, tsne, umap</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">names</span>(Pagwas<span class="sc">@</span>misc)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"Pathway_list"</span>                  <span class="st">"pca_cell_df"</span>                  </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>[<span class="dv">3</span>] <span class="st">"lm_results"</span>                    <span class="st">"bootstrap_results"</span>            </span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>[<span class="dv">5</span>] <span class="st">"scPathways_rankPvalue"</span>         <span class="st">"gene_heritability_correlation"</span></span></code></pre></div>
<p>Returns a Seruat data with entries(seruat_return=T):</p>
<ol style="list-style-type: decimal">
<li>
<p>In this Seruat result, two new assays were added:</p>
<ul>
<li><p><strong>scPagwasPaPca</strong>: An assay for S4 type of data; the svd score result for pathways in each cells;</p></li>
<li><p><strong>scPagwasPaHeritability</strong>: An assay for S4 type of data; the gPas score matrix for pathways in each cells;</p></li>
</ul>
</li>
</ol>
<!-- --><ol start="2" style="list-style-type: decimal">
<li>
<p>In meta.data, four columns were added:</p>
<ul>
<li><p><strong>scPagwas.TRS.Score1</strong>: inheritance related score, enrichment socre for genetics top genes;</p></li>
<li><p><strong>scPagwas.gPAS.score</strong>: Inheritance pathway regression effects score for each cells;</p></li>
<li><p><strong>CellScalepValue</strong>: Ranked CellScalepValue for each cells;</p></li>
<li><p><strong>CellScaleqValue</strong>: Ranked CellScaleqValue for each cells, adjust p value.</p></li>
</ul>
</li>
</ol>
<!-- --><ol start="3" style="list-style-type: decimal">
<li>
<p>A new element names misc is added in result, <code>Pagwas@misc</code> is a list data including:</p>
<ul>
<li><p><strong>Pathway_list</strong>: filtered pathway gene list;</p></li>
<li><p><strong>pca_cell_df</strong>: a pahtway and celltype data frame for pathway svd(1’st pca) result for each celltype;</p></li>
<li><p><strong>lm_results</strong>: the regression result for each celltype;</p></li>
<li><p><strong>gene_heritability_correlation</strong>: heritability correlation value for each gene;</p></li>
<li><p><strong>scPathways_rankPvalue</strong>: p values for each pathway in each single cell;</p></li>
<li><p><strong>bootstrap_results</strong>: The bootstrap data frame results for celltypes including bootstrap pvalue and confidence interval.</p></li>
</ul>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="pagwas-output-files-result">2.1 Pagwas : output files result<a class="anchor" aria-label="anchor" href="#pagwas-output-files-result"></a>
</h3>
<p>In <strong>monocytecount_bmmc</strong> result document folder, several result files are including:</p>
<ul>
<li><p><strong>scPagwas.run.log</strong> : The running time log for scPagwas;</p></li>
<li><p><strong>*_parameters.txt</strong> : The parameters log for scPagwas;</p></li>
<li><p><strong>*_cellytpes_bootstrap_results.csv</strong> : The result of cellytpes for bootstrap p results;</p></li>
<li><p><strong>*_gene_heritability_correlation.csv</strong> : The result of gene heritability correlation with gPAS score.</p></li>
<li><p><strong>*_Pathway_singlecell_lm_results.txt</strong> : The regression result for all pahtway and single cell matrix;</p></li>
<li><p>*_singlecell_Pathways_rankPvalue.csv : The pathway pvalue for eache singlecell;</p></li>
<li><p><strong>*_singlecell_scPagwas_score_pvalue.Result.csv</strong> : The data frame result for each cell inlcuding scPagwas.TRS.Score, scPagwas.gPAS.score, pValueHighScale, qValueHighScale;</p></li>
</ul>
<p><strong>scPagwas_cache</strong> is a temporary folder to save the SOAR data, you can remove it when finish the scPagwas. Sometimes, If scPagwas is failed of running for some reason, you should remove this folder or run this <code>SOAR::Remove(SOAR::Objects())</code> to remove the object in this folder.</p>
</div>
</div>
<div class="section level2">
<h2 id="result-visualization">3. Result Visualization<a class="anchor" aria-label="anchor" href="#result-visualization"></a>
</h2>
<div class="section level3">
<h3 id="visualize-the-scpagwas-score-of-single-cell-data-in-umap-or-tsne-plot-">3.1 Visualize the scPagwas Score of single cell data in UMAP or TSNE plot.<a class="anchor" aria-label="anchor" href="#visualize-the-scpagwas-score-of-single-cell-data-in-umap-or-tsne-plot-"></a>
</h3>
<ol style="list-style-type: decimal">
<li>DimPlot of singlecell data.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"RColorBrewer"</span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://satijalab.org/seurat" class="external-link">"Seurat"</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://satijalab.org/seurat" class="external-link">"SeuratObject"</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://nanx.me/ggsci/" class="external-link">"ggsci"</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">"ggpubr"</a></span><span class="op">)</span>
 <span class="co">#check the objects</span>
<span class="va">color26</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#D9DD6B"</span>,<span class="st">"#ECEFA4"</span>,<span class="st">"#D54C4C"</span>,<span class="st">"#8D2828"</span>,<span class="st">"#FDD2BF"</span>,<span class="st">"#E98580"</span>,<span class="st">"#DF5E5E"</span>,<span class="st">"#492F10"</span>,<span class="st">"#334257"</span>,<span class="st">"#476072"</span>,<span class="st">"#548CA8"</span>,
<span class="st">"#00A19D"</span>,<span class="st">"#ECD662"</span>,<span class="st">"#5D8233"</span>,<span class="st">"#284E78"</span>,<span class="st">"#3E215D"</span>,<span class="st">"#835151"</span>,<span class="st">"#F08FC0"</span>,<span class="st">"#C6B4CE"</span>,<span class="st">"#BB8760"</span>,<span class="st">"#FFDADA"</span>,<span class="st">"#3C5186"</span>,
<span class="st">"#558776"</span>,<span class="st">"#E99497"</span>,<span class="st">"#FFBD9B"</span>,<span class="st">"#0A1D37"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"monocyte_bmmc_dimplot_umap.png"</span>,width <span class="op">=</span> <span class="fl">600</span>, height <span class="op">=</span> <span class="fl">600</span><span class="op">)</span>
 <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">Pagwas</span>,pt.size<span class="op">=</span><span class="fl">1</span>,reduction<span class="op">=</span><span class="st">"tsne"</span>,label <span class="op">=</span> <span class="cn">T</span>, repel<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span>
 <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"celltypes"</span>, values <span class="op">=</span> <span class="va">color26</span><span class="op">)</span><span class="op">+</span>
 <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Monocyte BMMC"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"TSNE"</span>,y<span class="op">=</span><span class="st">""</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="figures/monocyte_bmmc_dimplot_umap.png" width="50%"></p>
<p>scPagwas.TRS.Score1 and positive(p&lt;0.05) cells showing in dimplot.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="../reference/scPagwas_Visualization.html">scPagwas_Visualization</a></span><span class="op">(</span>Single_data<span class="op">=</span><span class="va">Pagwas</span>,
                        p_thre <span class="op">=</span> <span class="fl">0.05</span>,
                        FigureType <span class="op">=</span> <span class="st">"tsne"</span>,
                        width <span class="op">=</span> <span class="fl">7</span>,
                        height <span class="op">=</span> <span class="fl">7</span>,
                        lowColor <span class="op">=</span> <span class="st">"white"</span>, 
                        highColor <span class="op">=</span> <span class="st">"red"</span>,
                        output.dirs<span class="op">=</span><span class="st">"figure"</span>,
                        size <span class="op">=</span> <span class="fl">0.5</span>,
                        do_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>scPagwas.gPAS.score dimplot.</li>
</ol>
<p><img src="figures/scPagwas.gPAS.score_tsne.png" width="40%"></p>
<ol start="3" style="list-style-type: decimal">
<li>scPagwas.TRS.Score dimplot.</li>
</ol>
<p><img src="figures/scPagwas.TRS.Score_tsne.png" width="40%"></p>
<ol start="4" style="list-style-type: decimal">
<li>scPagwas_CellScaleqValue dimplot.</li>
</ol>
<p><img src="figures/scPagwas_CellScaleqValue0.05_tsne.png" width="30%"></p>
<p>Positive cells(q value&lt;0.05): red dot; Negative cells: other dot.</p>
</div>
<div class="section level3">
<h3 id="plot-the-barplot-of-the-proportion-of-positive-cell-">3.2 Plot the barplot of the proportion of positive Cell.<a class="anchor" aria-label="anchor" href="#plot-the-barplot-of-the-proportion-of-positive-cell-"></a>
</h3>
<p>Plot the barplot of the proportion of positive Cells in celltypes:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_bar_positie_nagtive.html">plot_bar_positie_nagtive</a></span><span class="op">(</span>seurat_obj<span class="op">=</span><span class="va">Pagwas</span>,
                         var_ident<span class="op">=</span><span class="st">"celltypes"</span>,
                         var_group<span class="op">=</span><span class="st">"positiveCells"</span>,
                         vec_group_colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#E8D0B3"</span>,<span class="st">"#7EB5A6"</span><span class="op">)</span>,
                         do_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="figures/plot_bar_positie_nagtive.png" width="60%"></p>
<p>Plot the barplot of the proportion of celltypes in positive and negative Cells:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"plot_bar_positie_nagtive2.png"</span>,width <span class="op">=</span> <span class="fl">600</span>, height <span class="op">=</span> <span class="fl">600</span><span class="op">)</span>
<span class="fu"><a href="../reference/plot_bar_positie_nagtive.html">plot_bar_positie_nagtive</a></span><span class="op">(</span>seurat_obj<span class="op">=</span><span class="va">Pagwas</span>,
                              var_ident<span class="op">=</span><span class="st">"positiveCells"</span>,
                              var_group<span class="op">=</span><span class="st">"celltypes"</span>,
                              p_thre <span class="op">=</span> <span class="fl">0.01</span>,
                              vec_group_colors<span class="op">=</span><span class="cn">NULL</span>,
                              f_color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">10</span>, name<span class="op">=</span><span class="st">"RdYlBu"</span><span class="op">)</span><span class="op">)</span>,
                              do_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="figures/plot_bar_positie_nagtive2.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="visualize-the-heritability-correlated-pathways">3.3 Visualize the heritability correlated Pathways<a class="anchor" aria-label="anchor" href="#visualize-the-heritability-correlated-pathways"></a>
</h3>
<p>Plot the specific genetics pathway for each celltypes</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/grimbough/rhdf5" class="external-link">"rhdf5"</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grDevices</span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://factominer.free.fr" class="external-link">FactoMineR</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/andrie/ggdendro" class="external-link">ggdendro</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-forge.r-project.org/projects/grimport/" class="external-link">grImport2</a></span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span>
 <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mvkorpel/sisal" class="external-link">sisal</a></span><span class="op">)</span>

 <span class="kw"><a href="https://rdrr.io/r/base/source.html" class="external-link">source</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"plot_scpathway_contri_dot.R"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"plot_scpathway_dot.png"</span>,width <span class="op">=</span> <span class="fl">1000</span>, height <span class="op">=</span> <span class="fl">600</span><span class="op">)</span>
<span class="fu">plot_scpathway_dot</span><span class="op">(</span>Pagwas<span class="op">=</span><span class="va">Pagwas</span>,
                   celltypes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"01_HSC"</span>,<span class="st">"02_Early.Eryth"</span>,<span class="st">"05_CMP.LMPP"</span>,<span class="st">"11_CD14.Mono.1"</span>,<span class="st">"12_CD14.Mono.2"</span>,<span class="st">"13_CD16.Mono"</span>,<span class="st">"15_CLP.2"</span>,<span class="st">"19_CD8.N"</span>,<span class="st">"20_CD4.N1"</span>,<span class="st">"21_CD4.N2"</span><span class="op">)</span>,
                   topn_path_celltype<span class="op">=</span><span class="fl">5</span>,
                   filter_p<span class="op">=</span><span class="fl">0.05</span>,
                   max_logp<span class="op">=</span><span class="fl">15</span>,
                   display_max_sizes<span class="op">=</span><span class="cn">F</span>,
                   size_var <span class="op">=</span><span class="st">"logrankPvalue"</span> ,
                   col_var<span class="op">=</span><span class="st">"proportion"</span>,
                   shape.scale <span class="op">=</span> <span class="fl">8</span>,
                   cols.use<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lightgrey"</span>, <span class="st">"#E45826"</span><span class="op">)</span>,
                   dend_x_var <span class="op">=</span> <span class="st">"logrankPvalue"</span>,
                   dist_method<span class="op">=</span><span class="st">"euclidean"</span>,
                   hclust_method<span class="op">=</span><span class="st">"ward.D"</span>,
                   do_plot <span class="op">=</span> <span class="cn">T</span>,
                   <span class="co">#figurenames = "Pathway_plot.pdf",</span>
                   width <span class="op">=</span> <span class="fl">7</span>,
                   height <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="figures/plot_scpathway_dot.png" width="80%"></p>
</div>
<div class="section level3">
<h3 id="plot-the-heritability-correlated-genes">3.4 Plot the heritability correlated genes<a class="anchor" aria-label="anchor" href="#plot-the-heritability-correlated-genes"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"heritability_cor_scatterplot.pdf"</span><span class="op">)</span>
<span class="fu"><a href="../reference/heritability_cor_scatterplot.html">heritability_cor_scatterplot</a></span><span class="op">(</span>gene_heri_cor<span class="op">=</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">gene_heritability_correlation</span>,
                             topn_genes_label<span class="op">=</span><span class="fl">10</span>,
                             color_low<span class="op">=</span><span class="st">"#035397"</span>,
                             color_high <span class="op">=</span><span class="st">"#F32424"</span>,
                             color_mid <span class="op">=</span> <span class="st">"white"</span>,
                             text_size<span class="op">=</span><span class="fl">2</span>,
                             do_plot<span class="op">=</span><span class="cn">T</span>,
                             max.overlaps <span class="op">=</span><span class="fl">20</span>,
                             width <span class="op">=</span> <span class="fl">7</span>,
                             height <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="figures/heritability_cor_scatterplot.png" width="50%"></p>
</div>
<div class="section level3">
<h3 id="show-expression-of-the-top-heritability-correlation-genes-in-celltypes">3.5 Show expression of the top heritability correlation genes in celltypes<a class="anchor" aria-label="anchor" href="#show-expression-of-the-top-heritability-correlation-genes-in-celltypes"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://nanx.me/ggsci/" class="external-link">"ggsci"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://satijalab.org/seurat" class="external-link">"Seurat"</a></span><span class="op">)</span>
<span class="va">top5genes</span><span class="op">&lt;-</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">gene_heritability_correlation</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">gene_heritability_correlation</span>,decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"plot_vln_Corgenes.pdf"</span>,width <span class="op">=</span> <span class="fl">6</span>, height <span class="op">=</span><span class="fl">7</span><span class="op">)</span>
<span class="fu"><a href="../reference/plot_vln_Corgenes.html">plot_vln_Corgenes</a></span><span class="op">(</span>seurat_obj<span class="op">=</span><span class="va">Pagwas</span>,
             assay<span class="op">=</span><span class="st">"RNA"</span>, slot<span class="op">=</span><span class="st">"data"</span>,
             var_group<span class="op">=</span><span class="st">"celltypes"</span>,
             vec_features<span class="op">=</span><span class="va">top5genes</span>,
             vec_group_colors<span class="op">=</span> <span class="va">color26</span>,
             do_plot <span class="op">=</span> <span class="cn">T</span>
             <span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="figures/plot_vln_Corgenes.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="celltypes-bootstrap_results-reuslt">3.6 celltypes bootstrap_results reuslt<a class="anchor" aria-label="anchor" href="#celltypes-bootstrap_results-reuslt"></a>
</h3>
<p>Barplot for celltypes</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/Bootstrap_P_Barplot.html">Bootstrap_P_Barplot</a></span><span class="op">(</span>p_results<span class="op">=</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">bootstrap_results</span><span class="op">$</span><span class="va">bp_value</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>,
                    p_names<span class="op">=</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">bootstrap_results</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>,
                    figurenames <span class="op">=</span> <span class="st">"Bootstrap_P_Barplot.pdf"</span>,
                    width <span class="op">=</span> <span class="fl">5</span>,
                    height <span class="op">=</span> <span class="fl">7</span>,
                    do_plot<span class="op">=</span><span class="cn">T</span>,
                    title <span class="op">=</span> <span class="st">"monocytecount_bmmc"</span><span class="op">)</span></code></pre></div>
<p><img src="figures/Bootstrap_P_Barplot.png" width="40%"></p>
<p>Forest plot for estimate value.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"estimate_Plot.pdf"</span><span class="op">)</span>
<span class="fu"><a href="../reference/Bootstrap_estimate_Plot.html">Bootstrap_estimate_Plot</a></span><span class="op">(</span>Pagwas<span class="op">=</span><span class="va">Pagwas</span>,
                        width <span class="op">=</span> <span class="fl">9</span>,
                        height <span class="op">=</span> <span class="fl">7</span>,
                        do_plot<span class="op">=</span><span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="figures/estimate_Plot_00.png" width="60%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 10 x64 (build 19044)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_China.936 </span>
<span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_China.936   </span>
<span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_China.936</span>
<span class="co">#&gt; [4] LC_NUMERIC=C                              </span>
<span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_China.936    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] knitr_1.33</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] magrittr_2.0.1      R6_2.5.0            ragg_1.2.2         </span>
<span class="co">#&gt;  [4] rlang_0.4.11        fastmap_1.1.0       stringr_1.4.0      </span>
<span class="co">#&gt;  [7] tools_4.1.0         xfun_0.24           jquerylib_0.1.4    </span>
<span class="co">#&gt; [10] systemfonts_1.0.4   htmltools_0.5.1.1   yaml_2.2.1         </span>
<span class="co">#&gt; [13] digest_0.6.27       rprojroot_2.0.2     pkgdown_2.0.2      </span>
<span class="co">#&gt; [16] crayon_1.4.1        textshaping_0.3.6   BiocManager_1.30.16</span>
<span class="co">#&gt; [19] purrr_0.3.4         sass_0.4.0          fs_1.5.0           </span>
<span class="co">#&gt; [22] memoise_2.0.1       cachem_1.0.6        evaluate_0.14      </span>
<span class="co">#&gt; [25] rmarkdown_2.12      stringi_1.6.1       compiler_4.1.0     </span>
<span class="co">#&gt; [28] bslib_0.2.5.1       desc_1.3.0          jsonlite_1.7.2     </span>
<span class="co">#&gt; [31] renv_0.15.4</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chunyu Deng.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
