<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scPagwas">
<title>Conventional Parameters and Usage Instructions with Demo Example Data • scPagwas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Conventional Parameters and Usage Instructions with Demo Example Data">
<meta property="og:description" content="scPagwas">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scPagwas</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Conventional%20Parameters%20and%20Usage%20Instructions%20with%20Demo%20Example%20Data.html">Conventional Parameters and Usage Instructions with Demo Example Data</a>
    <a class="dropdown-item" href="../articles/Conventional%20result%20and%20visualization%20Instructions%20with%20Real-World%20Examples.html">Conventional result and visualization Instructions with Real-World Examples</a>
    <a class="dropdown-item" href="../articles/Introduction%20to%20Data%20Input%20and%20Preprocessing%20in%20scPagwas.html">Introduction to Data Input and Preprocessing in scPagwas</a>
    <a class="dropdown-item" href="../articles/Perform%20calculations%20for%20multiple%20traits%20based%20on%20a%20single-cell%20dataset.html">Perform calculations for multiple traits based on a single-cell dataset</a>
    <a class="dropdown-item" href="../articles/Pruning%20Process%20for%20GWAS%20Summary%20Statistics%20File%20in%20scPagwas.html">Pruning Process for GWAS Summary Statistics File in scPagwas</a>
    <a class="dropdown-item" href="../articles/Strategies%20for%20Large-scale%20Single-cell%20Data%20Subsetting%20and%20Computation.html">Strategies for Large-scale Single-cell Data Subsetting and Computation</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Conventional Parameters and Usage Instructions with Demo Example Data</h1>
            
            <h4 data-toc-skip class="date">Last Updated on 09 九月,
2023</h4>
      
      
      <div class="d-none name"><code>Conventional Parameters and Usage Instructions with Demo Example Data.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="paramers">1.Paramers<a class="anchor" aria-label="anchor" href="#paramers"></a>
</h2>
<p>Parameters and their Descriptions in scPagwas:</p>
<ul>
<li><p>Pagwas = NULL: This parameter is typically not required and does
not need any input data. When seurat_return = FALSE, all intermediate
data is stored in the “Pagwas” list and returned as the result. This
result can be inherited and used as input for subsequent calculations.
In certain scenarios, such as when performing two computations with the
same single-cell data input but different GWAS data inputs, the result
list obtained from the first computation can be used as the “Pagwas”
parameter input for the second computation. This allows skipping the
single-cell calculations, significantly expediting the process. However,
when seurat_return = TRUE, the returned result cannot be manipulated in
this manner, as it is the final Seurat result with many intermediate
data removed.</p></li>
<li><p><strong>gwas_data</strong> = NULL: The user needs to provide the
address or dataframe of the GWAS data. Specific formatting requirements
can be found in the “Introduction to Data Input and Preprocessing in
scPagwas”.</p></li>
<li><p>output.prefix = “Test”: This parameter sets the prefix for the
output result files.</p></li>
<li><p>output.dirs = “scPagwastest_output”: This parameter specifies the
directory for the output result files.</p></li>
<li><p>block_annotation = block_annotation: The genomic annotation file,
typically provided by scPagwas unless otherwise specified.</p></li>
<li><p><strong>Single_data</strong> = NULL: The single-cell data in
Seurat format, with preprocessed and annotated information stored in the
Idents. For specific processing steps, refer to the “Introduction to
Data Input and Preprocessing in scPagwas”.</p></li>
<li><p>assay = “RNA”: The name of the assay for the single-cell data
used in the computation.</p></li>
<li><p>Pathway_list = Genes_by_pathway_kegg: Pathway data, represented
as a list containing multiple pathways with their respective gene sets.
scPagwas provides various pathway data options, which can be found in
the “Introduction to Data Input and Preprocessing in scPagwas”.</p></li>
<li><p>chrom_ld = chrom_ld: LD (Linkage Disequilibrium) information
file, typically provided by scPagwas.</p></li>
<li><p>run_split = FALSE: This parameter is used for data subsetting and
computation. In scenarios where the server’s memory is insufficient to
handle large-scale single-cell data, the data can be split for
computation. The results are then integrated for further analysis. This
strategy will be further explained in the section titled “Strategies for
Large-scale Single-cell Data Subsetting and Computation”.</p></li>
<li><p>n.cores = 1: This parameter enables multi-core computation,
specifically in the sixth step of scPagwas, the
“Link_pathway_blocks_gwas” function.</p></li>
<li><p>marg = 10000: The boundary distance in base pairs from the
Transcription Start Site (TSS) position. The default is set to 10 kb,
resulting in a window size of 20 kb.</p></li>
<li><p>maf_filter = 0.01: Filtering threshold for Minor Allele Frequency
(MAF) in GWAS data. This step helps filter out SNPs, reducing the total
number of SNPs for computational efficiency. The MAF value has no impact
on the final results.</p></li>
<li><p>min_clustercells = 10: Filtering criterion for clusters in
single-cell annotation. Only clusters with a minimum number of cells
specified by this parameter will be considered.</p></li>
<li><p>min.pathway.size = 5: Filtering criterion for pathway data. Only
pathways with a gene count not less than 5 will be included in the
analysis.</p></li>
<li>
<p>max.pathway.size = 300: Filtering criterion for pathway data.
Only pathways with a gene count not greater than 300 will be included in
the analysis.</p>
<p>Note that during the computation, the filtering criteria ensure that
the number of expressed genes in a pathway is not less than 5 and not
more than 300. This helps prevent abnormal pathway calculations.
However, it means that for different single-cell datasets with the same
pathway data, the pathways included in the analysis may vary. The
filtered pathway data can be found in the “misc” section of the scPagwas
results.</p>
</li>
<li><p>iters_celltype = 200: Number of iterations for bootstrap
computation of cell-type p-values.</p></li>
<li><p>iters_singlecell = 100: Number of iterations for
background-corrected single-cell p-value computation. Note that this
parameter significantly affects the computation time.</p></li>
<li><p>n_topgenes = 1000: Number of top genes selected for calculating
the Transcriptional Regulator Score (TRS). Here, we follow the common
practice of selecting 1000 genes.</p></li>
<li><p>singlecell = TRUE: Whether to generate single-cell
results.</p></li>
<li><p>celltype = TRUE: Whether to generate cell-type results.</p></li>
<li><p>seurat_return = TRUE: Whether to output results in the Seurat
format. This parameter is only effective when singlecell =
TRUE.</p></li>
<li><p>remove_outlier = TRUE: Whether to filter out outliers.</p></li>
</ul>
<p>Please note that the above descriptions are for the provided
parameters in scPagwas.</p>
</div>
<div class="section level2">
<h2 id="scpagwas-calculation-processes-based-on-singlecell-and-celltype-parameters">2.scPagwas Calculation Processes Based on Singlecell and Celltype
Parameters<a class="anchor" aria-label="anchor" href="#scpagwas-calculation-processes-based-on-singlecell-and-celltype-parameters"></a>
</h2>
<p>There are some different situations for running scPagwas!</p>
<div class="section level3">
<h3 id="run-both-singlecell-and-celltypes-functions-">2.1. Run both singlecell and celltypes functions.<a class="anchor" aria-label="anchor" href="#run-both-singlecell-and-celltypes-functions-"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scPagwas</span><span class="op">)</span></span>
<span><span class="va">Pagwas</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/scPagwas_main.html">scPagwas_main</a></span><span class="op">(</span>Pagwas <span class="op">=</span><span class="cn">NULL</span>,</span>
<span>                     gwas_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GWAS_summ_example.txt"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                     Single_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNAexample.rds"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                     output.prefix<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                     output.dirs<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                     Pathway_list<span class="op">=</span><span class="va">Genes_by_pathway_kegg</span>,</span>
<span>                     assay<span class="op">=</span><span class="st">"RNA"</span>,</span>
<span>                     singlecell<span class="op">=</span><span class="cn">T</span>, </span>
<span>                     iters_singlecell <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                     celltype<span class="op">=</span><span class="cn">T</span>,</span>
<span>                     block_annotation <span class="op">=</span> <span class="va">block_annotation</span>,</span>
<span>                     chrom_ld <span class="op">=</span> <span class="va">chrom_ld</span><span class="op">)</span></span></code></pre></div>
<p>Interpretation of Results: Conventional Result and Visualization
Instructions with Real-World Examples.</p>
</div>
<div class="section level3">
<h3 id="only-run-celltypes-functions-">2.2. Only run celltypes functions.<a class="anchor" aria-label="anchor" href="#only-run-celltypes-functions-"></a>
</h3>
<p>The executive programs for celltypes and single cell are independent,
If you only want to know the celtypes, set the <code>celltype=T</code>
and <code>singlecell=F</code>. The advantages is save a lot of times,
for celltype only can omit many running processes.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas_celltypes</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/scPagwas_main.html">scPagwas_main</a></span><span class="op">(</span>Pagwas <span class="op">=</span><span class="cn">NULL</span>,</span>
<span>                     gwas_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GWAS_summ_example.txt"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                     Single_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNAexample.rds"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                     output.prefix<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                     output.dirs<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                     Pathway_list<span class="op">=</span><span class="va">Genes_by_pathway_kegg</span>,</span>
<span>                     assay<span class="op">=</span><span class="st">"RNA"</span>,</span>
<span>                     singlecell<span class="op">=</span><span class="cn">F</span>, </span>
<span>                     celltype<span class="op">=</span><span class="cn">T</span>,</span>
<span>                     block_annotation <span class="op">=</span> <span class="va">block_annotation</span>,</span>
<span>                     chrom_ld <span class="op">=</span> <span class="va">chrom_ld</span><span class="op">)</span></span></code></pre></div>
<p>The reuslt of celltypes is list format(not seurat format).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">names</span>(Pagwas_celltypes)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a> [<span class="dv">1</span>] <span class="st">"Celltype_anno"</span>     <span class="st">"data_mat"</span>          <span class="st">"VariableFeatures"</span>  <span class="st">"merge_scexpr"</span>     </span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a> [<span class="dv">5</span>] <span class="st">"rawPathway_list"</span>   <span class="st">"Pathway_list"</span>      <span class="st">"pca_scCell_mat"</span>    <span class="st">"pca_cell_df"</span>      </span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a> [<span class="dv">9</span>] <span class="st">"snp_gene_df"</span>       <span class="st">"lm_results"</span>        <span class="st">"bootstrap_results"</span></span></code></pre></div>
<p>Pagwas_celltypes: A Result List with Distinct Characteristics and
Inheritable Intermediate Data.</p>
<ul>
<li>bootstrap_results: The bootstrap data frame results for celltypes
including bootstrap pvalue and confidence interval.</li>
<li>pca_scCell_mat : a pahtway and cell data matrix for pathway svd(1’st
pca) result for each cell;</li>
<li>pca_cell_df : a pahtway and celltype data matrix for pathway
svd(1’st pca) result for each celltype;</li>
<li>Other elements are the intermediate data.</li>
</ul>
</div>
<div class="section level3">
<h3 id="only-run-singlecell-functions">2.3. Only run singlecell functions<a class="anchor" aria-label="anchor" href="#only-run-singlecell-functions"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas_singlecell</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/scPagwas_main.html">scPagwas_main</a></span><span class="op">(</span>Pagwas <span class="op">=</span><span class="cn">NULL</span>, </span>
<span>                                 gwas_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GWAS_summ_example.txt"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                                 Single_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNAexample.rds"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                                 output.prefix<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                                 output.dirs<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                                 Pathway_list<span class="op">=</span><span class="va">Genes_by_pathway_kegg</span>,</span>
<span>                                 assay<span class="op">=</span><span class="st">"RNA"</span>,</span>
<span>                                 singlecell<span class="op">=</span><span class="cn">T</span>, </span>
<span>                                 celltype<span class="op">=</span><span class="cn">F</span>,</span>
<span>                                 block_annotation <span class="op">=</span> <span class="va">block_annotation</span>,</span>
<span>                                 chrom_ld <span class="op">=</span> <span class="va">chrom_ld</span><span class="op">)</span></span></code></pre></div>
<p>Because the parameters and input data are the same with celltypes
function, we also can take the celltypes result as the input data for
single cell function. The advantage is there is no need to run the
<code>svd</code> code block, save a lot of time.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas_singlecell</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/scPagwas_main.html">scPagwas_main</a></span><span class="op">(</span>Pagwas <span class="op">=</span><span class="va">Pagwas_celltypes</span>, </span>
<span>                                 gwas_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GWAS_summ_example.txt"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                                 Single_data <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNAexample.rds"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span>,</span>
<span>                                 output.prefix<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                                 output.dirs<span class="op">=</span><span class="st">"Test"</span>,</span>
<span>                                 Pathway_list<span class="op">=</span><span class="va">Genes_by_pathway_kegg</span>,</span>
<span>                                 assay<span class="op">=</span><span class="st">"RNA"</span>,</span>
<span>                                 singlecell<span class="op">=</span><span class="cn">T</span>, </span>
<span>                                 celltype<span class="op">=</span><span class="cn">F</span>,</span>
<span>                                 block_annotation <span class="op">=</span> <span class="va">block_annotation</span>,</span>
<span>                                 chrom_ld <span class="op">=</span> <span class="va">chrom_ld</span><span class="op">)</span></span></code></pre></div>
<p>The result is seurat format.</p>
</div>
</div>
<div class="section level2">
<h2 id="running-scpagwas-step-by-step">3. Running scPagwas step by step<a class="anchor" aria-label="anchor" href="#running-scpagwas-step-by-step"></a>
</h2>
<p>The main function, scPagwas_main, is actually a package of multiple
sub-functions designed to simplify the process. However, in practical
calculations, a more flexible approach may be required. Here, we will
introduce each step one by one to better understand the entire
computational workflow.</p>
<p>We use an example provided by scPagwas package.</p>
<div class="section level3">
<h3 id="single-data-input">3.1 Single data input<a class="anchor" aria-label="anchor" href="#single-data-input"></a>
</h3>
<p>The first step involves the reading and preprocessing of single
cells, primarily aimed at filtering out clusters with very few cells and
obtaining the intersection of genes between Genes_by_pathway_kegg and
single-cell genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scPagwas</span><span class="op">)</span></span>
<span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">Single_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNAexample.rds"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Single_data_input.html">Single_data_input</a></span><span class="op">(</span></span>
<span>      Pagwas <span class="op">=</span> <span class="va">Pagwas</span>,</span>
<span>      assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>      Single_data <span class="op">=</span> <span class="va">Single_data</span>,</span>
<span>      Pathway_list <span class="op">=</span> <span class="va">Genes_by_pathway_kegg</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">Single_data</span> <span class="op">&lt;-</span> <span class="va">Single_data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">data_mat</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">)</span></span>
<span><span class="co">#[1] "Celltype_anno"    "data_mat"         "VariableFeatures" "merge_scexpr"</span></span></code></pre></div>
<ul>
<li>“Celltype_anno”: Cell type annotation information.</li>
<li>“data_mat”: Single-cell expression matrix.</li>
<li>“VariableFeatures”: Information on variable genes in single
cells.</li>
<li>“merge_scexpr”: Average expression matrix of cell types.</li>
</ul>
</div>
<div class="section level3">
<h3 id="run-pathway-pca-score">3.2 Run pathway pca score<a class="anchor" aria-label="anchor" href="#run-pathway-pca-score"></a>
</h3>
<p>The SVD method is used to compute the SVD results for single cells
and cell types.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Pathway_pcascore_run.html">Pathway_pcascore_run</a></span><span class="op">(</span></span>
<span>        Pagwas <span class="op">=</span> <span class="va">Pagwas</span>,</span>
<span>        Pathway_list <span class="op">=</span> <span class="va">Genes_by_pathway_kegg</span></span>
<span>      <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">)</span></span>
<span><span class="co">#[1] "Celltype_anno"    "data_mat"         "VariableFeatures" "merge_scexpr"     "rawPathway_list" </span></span>
<span><span class="co">#[6] "Pathway_list"     "pca_scCell_mat"   "pca_cell_df"</span></span></code></pre></div>
<ul>
<li>“rawPathway_list”: Raw pathway data input.</li>
<li>“Pathway_list”: Filtered pathway data.</li>
<li>“pca_scCell_mat”: SVD results of pathways for single cells, with
rows representing pathways and columns representing cells.</li>
<li>“pca_cell_df”: SVD results of pathways for cell types, with rows
representing pathways and columns representing cell types.</li>
</ul>
</div>
<div class="section level3">
<h3 id="gwas-summary-data-input">3.3 GWAS summary data input<a class="anchor" aria-label="anchor" href="#gwas-summary-data-input"></a>
</h3>
<p>Read the GWAS summary data, remove the MHC and sex chromosome and
filtered the maf of SNP.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwas_data</span> <span class="op">&lt;-</span> <span class="fu">bigreadr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bigreadr/man/fread2.html" class="external-link">fread2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GWAS_summ_example.txt"</span>, package <span class="op">=</span> <span class="st">"scPagwas"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWAS_summary_input.html">GWAS_summary_input</a></span><span class="op">(</span></span>
<span>    Pagwas <span class="op">=</span> <span class="va">Pagwas</span>,</span>
<span>    gwas_data <span class="op">=</span> <span class="va">gwas_data</span>,</span>
<span>    maf_filter <span class="op">=</span> <span class="fl">0.1</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">)</span></span>
<span><span class="co">#[1] "Celltype_anno"    "data_mat"         "VariableFeatures" "merge_scexpr"     "rawPathway_list" </span></span>
<span><span class="co">#[6] "Pathway_list"     "pca_scCell_mat"   "pca_cell_df"      "gwas_data"</span></span></code></pre></div>
<ul>
<li>“gwas_data”：Gwas data input.</li>
</ul>
</div>
<div class="section level3">
<h3 id="mapping-snps-to-genes">3.4 Mapping Snps to Genes<a class="anchor" aria-label="anchor" href="#mapping-snps-to-genes"></a>
</h3>
<p>We set the <code>marg</code> is 10KB, means the position for SNP is
less 10KB distance to TSS of gene.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas</span><span class="op">$</span><span class="va">snp_gene_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SnpToGene.html">SnpToGene</a></span><span class="op">(</span></span>
<span>        gwas_data <span class="op">=</span> <span class="va">Pagwas</span><span class="op">$</span><span class="va">gwas_data</span>,</span>
<span>        block_annotation <span class="op">=</span> <span class="va">block_annotation</span>,</span>
<span>        marg <span class="op">=</span> <span class="fl">10000</span></span>
<span>      <span class="op">)</span></span></code></pre></div>
<ul>
<li>snp_gene_df: Correspondence between SNPs and genes.</li>
</ul>
</div>
<div class="section level3">
<h3 id="pathway-snp-annotation">3.5 Pathway-SNP annotation<a class="anchor" aria-label="anchor" href="#pathway-snp-annotation"></a>
</h3>
<p>Mapping SNPs to pathways and getting block data.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Pathway_annotation_input.html">Pathway_annotation_input</a></span><span class="op">(</span></span>
<span>      Pagwas <span class="op">=</span> <span class="va">Pagwas</span>,</span>
<span>      block_annotation <span class="op">=</span> <span class="va">block_annotation</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">)</span></span>
<span><span class="co">#[1] "Celltype_anno"    "data_mat"         "VariableFeatures" "merge_scexpr"     "rawPathway_list" </span></span>
<span><span class="co"># [6] "Pathway_list"     "pca_scCell_mat"   "pca_cell_df"      "gwas_data"        "snp_gene_df"     </span></span>
<span><span class="co">#[11] "pathway_blocks" </span></span></code></pre></div>
<ul>
<li>pathway_blocks: Each pathway is treated as a block, showing the
correspondence between genes and SNPs for each pathway.</li>
</ul>
</div>
<div class="section level3">
<h3 id="link-the-pathway-blocks-to-pca-score">3.6 Link the pathway blocks to pca score<a class="anchor" aria-label="anchor" href="#link-the-pathway-blocks-to-pca-score"></a>
</h3>
<p>The regression analysis step requires a relatively long computation
time.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Link_pathway_blocks_gwas.html">Link_pathway_blocks_gwas</a></span><span class="op">(</span></span>
<span>      Pagwas <span class="op">=</span> <span class="va">Pagwas</span>,</span>
<span>      chrom_ld <span class="op">=</span> <span class="va">chrom_ld</span>,</span>
<span>      singlecell <span class="op">=</span> <span class="cn">T</span>,</span>
<span>      celltype <span class="op">=</span> <span class="cn">T</span>,</span>
<span>      backingpath<span class="op">=</span><span class="st">"./temp"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">)</span></span>
<span> <span class="co">#[1] "Celltype_anno"        "data_mat"             "VariableFeatures"     "merge_scexpr"        </span></span>
<span> <span class="co">#[5] "rawPathway_list"      "Pathway_list"         "pca_scCell_mat"       "pca_cell_df"         </span></span>
<span> <span class="co">#[9] "snp_gene_df"          "Pathway_sclm_results" "Pathway_ld_gwas_data"</span></span></code></pre></div>
<ul>
<li><p>“Pathway_sclm_results”: Matrix of regression parameters for
single cells, with rows representing cells and columns representing
pathways.</p></li>
<li>
<p>“Pathway_ld_gwas_data”: Intermediate results of the regression
calculation, including x and y information for each pathway.</p>
<p>In this step, a temporary file will be generated in the result folder
to store intermediate data during the regression analysis, aiming to
improve efficiency. Although the contents of this file will be deleted
during the execution, sometimes it may not be completely removed. It is
recommended to manually delete it after the program finishes
running.</p>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="perform-regression-for-celltypes">3.7 Perform regression for celltypes<a class="anchor" aria-label="anchor" href="#perform-regression-for-celltypes"></a>
</h3>
<p>Run the regression function for celltypes.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas</span><span class="op">$</span><span class="va">lm_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Pagwas_perform_regression.html">Pagwas_perform_regression</a></span><span class="op">(</span>Pathway_ld_gwas_data <span class="op">=</span> <span class="va">Pagwas</span><span class="op">$</span><span class="va">Pathway_ld_gwas_data</span><span class="op">)</span></span>
<span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Boot_evaluate.html">Boot_evaluate</a></span><span class="op">(</span><span class="va">Pagwas</span>, bootstrap_iters <span class="op">=</span> <span class="fl">200</span>, part <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">)</span></span>
<span><span class="co"># [1] "Celltype_anno"        "data_mat"             "VariableFeatures"     "merge_scexpr"        </span></span>
<span><span class="co"># [5] "rawPathway_list"      "Pathway_list"         "pca_scCell_mat"       "pca_cell_df"         </span></span>
<span><span class="co"># [9] "snp_gene_df"          "Pathway_sclm_results" "Pathway_ld_gwas_data" "lm_results"          </span></span>
<span><span class="co">#[13] "bootstrap_results"</span></span>
<span><span class="co">#remove the Pathway_ld_gwas_data, it takes a lot of memory.</span></span>
<span><span class="va">Pagwas</span><span class="op">$</span><span class="va">Pathway_ld_gwas_data</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
<ul>
<li>“lm_results”: Regression results for cell types.</li>
<li>“bootstrap_results”: p-value results of bootstrap random testing for
cell types.</li>
</ul>
</div>
<div class="section level3">
<h3 id="construct-the-scpagwas-score">3.8 Construct the scPagwas score<a class="anchor" aria-label="anchor" href="#construct-the-scpagwas-score"></a>
</h3>
<p>The gPAS scPagwas score mainly to deal with the single-cell
regression result.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Pagwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scPagwas_perform_score.html">scPagwas_perform_score</a></span><span class="op">(</span></span>
<span>      Pagwas <span class="op">=</span> <span class="va">Pagwas</span>,</span>
<span>      remove_outlier <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">)</span></span>
<span><span class="co"># [1] "Celltype_anno"          "data_mat"               "VariableFeatures"      </span></span>
<span><span class="co"># [4] "merge_scexpr"           "rawPathway_list"        "Pathway_list"          </span></span>
<span><span class="co"># [7] "pca_scCell_mat"         "pca_cell_df"            "snp_gene_df"           </span></span>
<span><span class="co">#[10] "Pathway_sclm_results"   "Pathway_ld_gwas_data"   "lm_results"            </span></span>
<span><span class="co">#[13] "bootstrap_results"      "Pathway_single_results" "scPathways_rankPvalue" </span></span>
<span><span class="co">#[16] "scPagwas.gPAS.score" </span></span></code></pre></div>
<ul>
<li>“Pathway_single_results”: Regression results matrix, with rows
representing pathways and columns representing cells. The values in the
matrix represent the genetic contributions of different pathways in each
cell.</li>
<li>“scPathways_rankPvalue”: Rows represent pathways, and columns
represent cell types. The p-value indicates the significance level of
the genetic contribution of that pathway in each cell type.</li>
<li>“scPagwas.gPAS.score”: Genetic score for each cell, obtained by
summing the columns of “Pathway_single_results”. It represents the
overall contribution of all pathways.</li>
</ul>
</div>
<div class="section level3">
<h3 id="get-the-gene-heritability-correlation">3.9 Get the gene heritability correlation<a class="anchor" aria-label="anchor" href="#get-the-gene-heritability-correlation"></a>
</h3>
<p>Run heritability correlation for all genes.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#pcc gene for all gPas score!</span></span>
<span><span class="va">Pagwas</span><span class="op">$</span><span class="va">PCC</span> <span class="op">&lt;-</span> <span class="fu">scPagwas</span><span class="fu">::</span><span class="fu"><a href="../reference/scGet_PCC.html">scGet_PCC</a></span><span class="op">(</span>scPagwas.gPAS.score<span class="op">=</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">scPagwas.gPAS.score</span>,</span>
<span>                                    data_mat<span class="op">=</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">data_mat</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>PCC: Pearson correlation coefficients, Correlation index between
each gene and the genetic score.</li>
</ul>
</div>
<div class="section level3">
<h3 id="calculate-the-trs-score-for-top-genes-">3.10 Calculate the TRS score for top genes.<a class="anchor" aria-label="anchor" href="#calculate-the-trs-score-for-top-genes-"></a>
</h3>
<p>Calculate the TRS score for top genes by <code>AddModuleScore</code>
and running the p value for each cell by
<code>Get_CorrectBg_p</code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Obtain the top 500 genes with the highest PCC.</span></span>
<span><span class="va">n_topgenes</span><span class="op">=</span><span class="fl">500</span></span>
<span><span class="va">scPagwas_topgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">PCC</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">PCC</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_topgenes</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">scPagwas_downgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">PCC</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">PCC</span>, decreasing <span class="op">=</span><span class="cn">F</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_topgenes</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#Single_data refers to the single-cell data initially inputted.</span></span>
<span><span class="va">Single_data</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/AddModuleScore.html" class="external-link">AddModuleScore</a></span><span class="op">(</span><span class="va">Single_data</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">scPagwas_topgenes</span>,<span class="va">scPagwas_downgenes</span><span class="op">)</span>, name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"scPagwas.TRS.Score"</span>,<span class="st">"scPagwas.downTRS.Score"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Calculate the p-values for scPagwas.TRS.Score of single cells after background correction.</span></span>
<span><span class="va">correct_pdf</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/Get_CorrectBg_p.html">Get_CorrectBg_p</a></span><span class="op">(</span>Single_data<span class="op">=</span><span class="va">Single_data</span>,</span>
<span>                             scPagwas.TRS.Score<span class="op">=</span><span class="va">Single_data</span><span class="op">$</span><span class="va">scPagwas.TRS.Score1</span>,</span>
<span>                             iters_singlecell<span class="op">=</span><span class="fl">100</span>,</span>
<span>                             n_topgenes<span class="op">=</span><span class="fl">1000</span>,</span>
<span>                             scPagwas_topgenes<span class="op">=</span><span class="va">scPagwas_topgenes</span><span class="op">)</span></span>
<span><span class="va">Pagwas</span><span class="op">$</span><span class="va">Random_Correct_BG_pdf</span> <span class="op">&lt;-</span> <span class="va">correct_pdf</span></span>
<span></span>
<span><span class="co">#Merge the p-values of cells belonging to the same cell type into a single p-value for each cell type.</span></span>
<span><span class="va">Pagwas</span><span class="op">$</span><span class="va">Merged_celltype_pvalue</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/Merge_celltype_p.html">Merge_celltype_p</a></span><span class="op">(</span>single_p<span class="op">=</span><span class="va">correct_pdf</span><span class="op">$</span><span class="va">pooled_p</span>,celltype<span class="op">=</span><span class="va">Pagwas</span><span class="op">$</span><span class="va">Celltype_anno</span><span class="op">$</span><span class="va">annotation</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Integrate and output the results of single-cell analysis.</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>     scPagwas.TRS.Score <span class="op">=</span> <span class="va">Single_data</span><span class="op">$</span><span class="va">scPagwas.TRS.Score1</span>,</span>
<span>    scPagwas.downTRS.Score <span class="op">=</span> <span class="va">Single_data</span><span class="op">$</span><span class="va">scPagwas.downTRS.Score2</span>,</span>
<span>     scPagwas.gPAS.score <span class="op">=</span> <span class="va">Pagwas</span><span class="op">$</span><span class="va">scPagwas.gPAS.score</span>,</span>
<span>     Random_Correct_BG_p <span class="op">=</span> <span class="va">correct_pdf</span><span class="op">$</span><span class="va">pooled_p</span>,</span>
<span>     Random_Correct_BG_adjp <span class="op">=</span> <span class="va">correct_pdf</span><span class="op">$</span><span class="va">adj_p</span>,</span>
<span>     Random_Correct_BG_z <span class="op">=</span> <span class="va">correct_pdf</span><span class="op">$</span><span class="va">pooled_z</span><span class="op">)</span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">a</span>,file <span class="op">=</span> <span class="st">"_singlecell_scPagwas_score_pvalue.Result.csv"</span>,quote <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<ul>
<li><p>Single_data$scPagwas.TRS.Score1 : Trait relevant score based on
top trait relevant genes.</p></li>
<li>
<p>Single_data$scPagwas.downTRS.Score2: Trait relevant score based
on anti-relevant genes.</p>
<p>Note: Although our article primarily focuses on the TRS scores, it
does not imply that reverse “downTRS” are always devoid of significance.
During our following research, we observed that in certain phenotypes,
such as some cancer traits.When observing the density distribution plot
of gPas scores, it becomes evident that it exhibits a <strong>high and
sharp peak</strong>. This indicates that the majority of cells have
minor effects. However, if the distribution shows a more pronounced
long-tail behavior, the direction of this long tail can influence the
impact of PCC gene pairs on the phenotype.</p>
<div class="float">
<img src="figures/image-20230823150635803.png" alt="image-20230823150635803"><div class="figcaption">image-20230823150635803</div>
</div>
<p>In other words, specific genotypes may be more closely related to a
reduced risk of the phenotype. However, It requires a case-by-case
analysis. The question then arises: do GWAS genetic effect mainly have a
positive or negative impact on phenotypes? What factors determine this
impact? This is a highly intriguing question that we intend to explore
further in subsequent research. We encourage researchers to consider the
distribution of scPagwas.gPAS.score as an adjunctive factor for
assessing directionality, though it is not absolute and should be
analyzed on a case-by-case basis.</p>
</li>
<li><p>Random_Correct_BG_pdf : Dataframe containing the calculated
results of p-values after background correction for TRS of each cell,
including pooled_p, adj_p, pooled_z, and other results.</p></li>
<li><p>Merged_celltype_pvalue : P-value results of cell types after
merging the p-values of single cells.</p></li>
</ul>
<p>All these sub-functions for scPagwas can running dependently, but
need to run orderly. The <code>scPagwas_main</code> function is a
wrapper functions for these sub-functions.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chunyu Deng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
