<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scPagwas">
<title>Conventional result and visualization Instructions with Real-World Examples • scPagwas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Conventional result and visualization Instructions with Real-World Examples">
<meta property="og:description" content="scPagwas">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scPagwas</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Conventional%20Parameters%20and%20Usage%20Instructions%20with%20Demo%20Example%20Data.html">Conventional Parameters and Usage Instructions with Demo Example Data</a>
    <a class="dropdown-item" href="../articles/Conventional%20result%20and%20visualization%20Instructions%20with%20Real-World%20Examples.html">Conventional result and visualization Instructions with Real-World Examples</a>
    <a class="dropdown-item" href="../articles/Introduction%20to%20Data%20Input%20and%20Preprocessing%20in%20scPagwas.html">Introduction to Data Input and Preprocessing in scPagwas</a>
    <a class="dropdown-item" href="../articles/Perform%20calculations%20for%20multiple%20traits%20based%20on%20a%20single-cell%20dataset.html">Perform calculations for multiple traits based on a single-cell dataset</a>
    <a class="dropdown-item" href="../articles/Pruning%20Process%20for%20GWAS%20Summary%20Statistics%20File%20in%20scPagwas.html">Pruning Process for GWAS Summary Statistics File in scPagwas</a>
    <a class="dropdown-item" href="../articles/Strategies%20for%20Large-scale%20Single-cell%20Data%20Subsetting%20and%20Computation.html">Strategies for Large-scale Single-cell Data Subsetting and Computation</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Conventional result and visualization Instructions with Real-World Examples</h1>
            
            <h4 data-toc-skip class="date">Last Updated: 08, 八月, 2023
at 16:58</h4>
      
      
      <div class="d-none name"><code>Conventional result and visualization Instructions with Real-World Examples.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="utilizing-real-world-examples-in-scpagwas-for-comprehensive-testing">1.Utilizing Real-World Examples in scPagwas for Comprehensive
Testing<a class="anchor" aria-label="anchor" href="#utilizing-real-world-examples-in-scpagwas-for-comprehensive-testing"></a>
</h2>
<p>To showcase the visualization effects accurately, we have opted to
use real-world examples from the referenced article instead of the
software’s provided demo dataset. In this scenario, we will be utilizing
the BMMC scRNA-seq data in conjunction with the GWAS data for monocyte
count. It is important to note that due to the large size of the data,
the computation time for this analysis may take several minutes. Hence,
we recommend directly downloading the result data for convenient
examination and exploration. The processed monocytecount gwas data can
be download from <a href="https://drive.google.com/file/d/1drlkNcU9QA_lGDiJZlK42DRpb0ac4e-k/view?usp=sharing" class="external-link">here</a>.</p>
<p>BMMC example scRNA-seq data can be obtained from <a href="https://drive.google.com/file/d/1edkAtU5K1ljSyu2KMj5Q1ZVW9TxbqKgS/view?usp=sharing" class="external-link">here</a>.</p>
<p>We can download the result files from <a href="https://drive.google.com/file/d/1VUg9s4e-UhXL-XrKiBzTgdiBbKpX7AG1/view?usp=sharing" class="external-link">here</a></p>
<p>If you want to run the deno data, please go to <strong>Conventional
Usage Instructions for scPagwas with Demo Example Data</strong></p>
</div>
<div class="section level2">
<h2 id="computing-the-singlecell-and-celltype-result-for-monocytecount-trait">2. Computing the singlecell and celltype result for Monocytecount
trait<a class="anchor" aria-label="anchor" href="#computing-the-singlecell-and-celltype-result-for-monocytecount-trait"></a>
</h2>
<p>If we running scPagwas in multi-core in server environment, there may
cause an error:
<code>Error: Two levels of parallelism are used. See ?assert_cores\</code>`
add this code before call in R environment:</p>
<p><code>export OPENBLAS_NUM_THREADS=1</code></p>
<p>There is no need to run this code in window system.</p>
<p>In this example, we run the scPagwas for usual step, both running
singlecell and celltype process.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scPagwas</span><span class="op">)</span></span>
<span><span class="va">Pagwas</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/scPagwas_main.html">scPagwas_main</a></span><span class="op">(</span>Pagwas <span class="op">=</span><span class="cn">NULL</span>,</span>
<span>                     gwas_data<span class="op">=</span><span class="st">"D:/OneDrive/GWAS_Multiomics/gwas_data/monocytecount_prune_gwas_data.txt"</span>,</span>
<span>                     Single_data <span class="op">=</span><span class="st">"D:/OneDrive/GWAS_Multiomics/Compare/Hema_test2/Seu_Hema_data.rds"</span>,</span>
<span>                     output.prefix<span class="op">=</span><span class="st">"mono"</span>,</span>
<span>                     output.dirs<span class="op">=</span><span class="st">"monocytecount_bmmc"</span>,</span>
<span>                     Pathway_list<span class="op">=</span><span class="va">Genes_by_pathway_kegg</span>,</span>
<span>                     n.cores<span class="op">=</span><span class="fl">2</span>,</span>
<span>                     assay<span class="op">=</span><span class="st">"RNA"</span>,</span>
<span>                     singlecell<span class="op">=</span><span class="cn">T</span>, </span>
<span>                     iters_singlecell <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                     celltype<span class="op">=</span><span class="cn">T</span>,</span>
<span>                     block_annotation <span class="op">=</span> <span class="va">block_annotation</span>,</span>
<span>                     chrom_ld <span class="op">=</span> <span class="va">chrom_ld</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">Pagwas</span>,file<span class="op">=</span><span class="st">"./BMMC_scPagwas.RData"</span><span class="op">)</span></span></code></pre></div>
<p>Sometimes, we need to remove the objects in cache folder:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu">SOAR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SOAR/man/Objects.html" class="external-link">Objects</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">{</span></span>
<span> <span class="fu">SOAR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SOAR/man/Remove.html" class="external-link">Remove</a></span><span class="op">(</span><span class="fu">SOAR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SOAR/man/Objects.html" class="external-link">Objects</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="result-interpretation">3. Result interpretation<a class="anchor" aria-label="anchor" href="#result-interpretation"></a>
</h2>
<p>Data Results and Output in scPagwas:</p>
<ul>
<li>By default (seurat_return = TRUE), scPagwas returns the data results
in Seurat format, which ensures consistency with the input single-cell
data format. The scPagwas results in Seurat format include two
additional assays, five new columns in the meta.data, and five new
columns in the misc data.</li>
<li>In the output folder, there will be five result files along with an
empty folder named “scPagwas_cache”. Generally, the “scPagwas_cache”
folder is unnecessary and can be deleted without affecting the
results.</li>
</ul>
<div class="section level3">
<h3 id="pagwas">3.1 Pagwas<a class="anchor" aria-label="anchor" href="#pagwas"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">print</span>(Pagwas)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>An object of class Seurat </span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="dv">16478</span> features across <span class="dv">35582</span> samples within <span class="dv">3</span> assays </span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>Active assay<span class="sc">:</span> <span class="fu">RNA</span> (<span class="dv">15860</span> features, <span class="dv">5000</span> variable features)</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a> <span class="dv">2</span> other assays present<span class="sc">:</span> scPagwasPaHeritability, scPagwasPaPca</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a> <span class="dv">3</span> dimensional reductions calculated<span class="sc">:</span> pca, tsne, umap</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">names</span>(Pagwas<span class="sc">@</span>misc)</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"Pathway_list"</span>                  <span class="st">"pca_cell_df"</span>                   <span class="st">"lm_results"</span>                   </span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>[<span class="dv">4</span>] <span class="st">"bootstrap_results"</span>             <span class="st">"scPathways_rankPvalue"</span>         <span class="st">"GeneticExpressionIndex"</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>[<span class="dv">7</span>] <span class="st">"Merged_celltype_pvalue"</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>
<p>In this Seruat result, two new assays were added:</p>
<ul>
<li><p><strong>scPagwasPaPca</strong>: An assay for S4 type of data; the
svd score result for pathways in each cells;</p></li>
<li><p><strong>scPagwasPaHeritability</strong>: An assay for S4 type of
data; the gPas score matrix for pathways in each cells;</p></li>
</ul>
</li>
</ol>
<!-- --><p>These two files are intermediate files generated by scPagwas.</p>
<ol start="2" style="list-style-type: decimal">
<li>
<p>In meta.data, four columns were added:</p>
<ul>
<li>
<strong>scPagwas.TRS.Score1</strong>: inheritance related score,
enrichment socre for genetics top genes;</li>
</ul>
</li>
</ol>
<ul>
<li>
<strong>scPagwas.gPAS.score</strong>: Inheritance pathway regression
effects score for each cells;
<ul>
<li>
<strong>Random_Correct_BG_p</strong>: CellpValue for each
cells;</li>
</ul>
</li>
<li>
<strong>Random_Correct_BG_adjp</strong>: fdr for each cells, adjust
p value.
<ul>
<li>
<strong>Random_Correct_BG_z</strong>：z score for eahc cells.</li>
</ul>
</li>
</ul>
<!-- --><ol start="3" style="list-style-type: decimal">
<li>
<p>A new element names misc is added in scPagwas’s result,
<code>Pagwas@misc</code> is a list including:</p>
<ul>
<li><p><strong>Pathway_list</strong>: filtered pathway gene
list;</p></li>
<li><p><strong>pca_cell_df</strong>: a pahtway and celltype data frame
for pathway svd(1’st pca) result for each celltype;</p></li>
<li><p><strong>lm_results</strong>: the regression result for each
celltype;</p></li>
<li><p><strong>GeneticExpressionIndex</strong>: heritability correlation
value for each gene;</p></li>
<li><p><strong>bootstrap_results</strong>: The bootstrap data frame
results for celltypes including bootstrap pvalue and confidence
interval.</p></li>
<li>
<p><strong>Merged_celltype_pvalue</strong>: The resulting
<strong>Merged_celltype_pvalue</strong> represents p-values at the cell
type level, which differ from the <strong>bootstrap_results</strong>.
The bootstrap_results are based on merging cell-level expression data
into cell type-level expression data and performing calculations. In
contrast, the <strong>Merged_celltype_pvalue</strong> is derived from
the <strong>Random_Correct_BG_p</strong> results, combining the p-values
of all cells within a specific cell type into a single p-value result
for that cell type.</p>
<p><strong>Note</strong>: Although both the bootstrap and merged
strategies yield cell type-level results, they may not be identical.
This is because the bootstrap approach calculates p-values based on
pseudo-bulk data, which can differ from the results obtained from
single-cell calculations. In contrast, the merged strategy directly
integrates the results from single-cell calculations, resulting in
greater consistency with single-cell results. The primary focus in
scPagwas paper is on the bootstrap_results; however, during the later
stages of development, we discovered that discrepancies between cell
type-level and single-cell-level calculation strategies posed some
challenges in downstream analyses. Hence, in the new version
development, we directly utilize the integration of single-cell results
to obtain cell type-level p-values. Additionally, the major advantage of
the bootstrap_results is its speed, as the calculation of
Random_Correct_BG_p for single cells requires at least 100 iterations,
making it computationally demanding. To avoid calculating
Random_Correct_BG_p, setting the parameter iters_singlecell=0 is a good
option.</p>
</li>
</ul>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="pagwas-output-files-result">3.2 Pagwas : output files result<a class="anchor" aria-label="anchor" href="#pagwas-output-files-result"></a>
</h3>
<div class="float">
<img src="figures/resultfiles.png" alt="image-20230526142322858"><div class="figcaption">image-20230526142322858</div>
</div>
<p>In <strong>monocytecount_bmmc</strong> folder, There is several
result files :</p>
<ul>
<li><p><strong>*_cellytpes_bootstrap_results.csv</strong> : The result
of cellytpes for bootstrap p results;</p></li>
<li><p><strong>*_GeneticExpressionIndex.csv</strong> : The result of
gene heritability correlation with gPAS score.</p></li>
<li><p><strong>*_Pathway_singlecell_lm_results.txt</strong> : The
regression result for all pahtway and single cell matrix;</p></li>
<li><p><strong>*_singlecell_scPagwas_score_pvalue.Result.csv</strong> :
The data frame result for each cell inlcuding scPagwas.TRS.Score,
scPagwas.gPAS.score, pValueHighScale, qValueHighScale;</p></li>
<li><p><strong>*_Merged_celltype_pvalue.csv</strong> : The merged
singlecell pvalue to celltype pvalue.</p></li>
</ul>
<p><strong>scPagwas_cache</strong> is a temporary folder to save the
SOAR data, you can remove it when finish the scPagwas.</p>
</div>
</div>
<div class="section level2">
<h2 id="result-visualization">4. Result Visualization<a class="anchor" aria-label="anchor" href="#result-visualization"></a>
</h2>
<div class="section level3">
<h3 id="visualize-the-scpagwas-score-of-single-cell-data-in-umap-or-tsne-plot-">4.1 Visualize the scPagwas Score of single cell data in UMAP or TSNE
plot.<a class="anchor" aria-label="anchor" href="#visualize-the-scpagwas-score-of-single-cell-data-in-umap-or-tsne-plot-"></a>
</h3>
<ol style="list-style-type: decimal">
<li>DimPlot of singlecell data.</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"RColorBrewer"</span><span class="op">)</span></span>
<span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://satijalab.org/seurat" class="external-link">"Seurat"</a></span><span class="op">)</span></span>
<span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://mojaveazure.github.io/seurat-object/" class="external-link">"SeuratObject"</a></span><span class="op">)</span></span>
<span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://nanx.me/ggsci/" class="external-link">"ggsci"</a></span><span class="op">)</span></span>
<span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">"ggpubr"</a></span><span class="op">)</span></span>
<span> <span class="co">#check the objects</span></span>
<span><span class="va">color26</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#D9DD6B"</span>,<span class="st">"#ECEFA4"</span>,<span class="st">"#D54C4C"</span>,<span class="st">"#8D2828"</span>,<span class="st">"#FDD2BF"</span>,<span class="st">"#E98580"</span>,<span class="st">"#DF5E5E"</span>,<span class="st">"#492F10"</span>,<span class="st">"#334257"</span>,<span class="st">"#476072"</span>,<span class="st">"#548CA8"</span>,<span class="st">"#00A19D"</span>,<span class="st">"#ECD662"</span>,<span class="st">"#5D8233"</span>,<span class="st">"#284E78"</span>,<span class="st">"#3E215D"</span>,<span class="st">"#835151"</span>,<span class="st">"#F08FC0"</span>,<span class="st">"#C6B4CE"</span>,<span class="st">"#BB8760"</span>,<span class="st">"#FFDADA"</span>,<span class="st">"#3C5186"</span>,<span class="st">"#558776"</span>,<span class="st">"#E99497"</span>,<span class="st">"#FFBD9B"</span>,<span class="st">"#0A1D37"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">Pagwas</span>,pt.size<span class="op">=</span><span class="fl">1</span>,reduction<span class="op">=</span><span class="st">"tsne"</span>,label <span class="op">=</span> <span class="cn">T</span>, repel<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span></span>
<span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"celltypes"</span>, values <span class="op">=</span> <span class="va">color26</span><span class="op">)</span><span class="op">+</span></span>
<span> <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Monocyte BMMC"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"TSNE"</span>,y<span class="op">=</span><span class="st">""</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/monocyte_bmmc_dimplot_umap.png" width="50%"></p>
<p>scPagwas.TRS.Score1 and positive(p&lt;0.05) cells showing in
dimplot.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="fu"><a href="../reference/scPagwas_Visualization.html">scPagwas_Visualization</a></span><span class="op">(</span>Single_data<span class="op">=</span><span class="va">Pagwas</span>,</span>
<span>                        p_thre <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                        FigureType <span class="op">=</span> <span class="st">"tsne"</span>,</span>
<span>                        width <span class="op">=</span> <span class="fl">7</span>,</span>
<span>                        height <span class="op">=</span> <span class="fl">7</span>,</span>
<span>                        lowColor <span class="op">=</span> <span class="st">"white"</span>, </span>
<span>                        highColor <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>                        output.dirs<span class="op">=</span><span class="st">"figure"</span>,</span>
<span>                        size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                        do_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>scPagwas.gPAS.score dimplot.</li>
</ol>
<p><img src="figures/scPagwas.gPAS.score_tsne.png" width="40%"></p>
<ol start="3" style="list-style-type: decimal">
<li>scPagwas.TRS.Score dimplot.</li>
</ol>
<p><img src="figures/scPagwas.TRS.Score_tsne.png" width="40%"></p>
<ol start="4" style="list-style-type: decimal">
<li>scPagwas_CellScaleqValue dimplot.</li>
</ol>
<p><img src="figures/scPagwas_CellScaleqValue0.05_tsne.png" width="30%"></p>
<p>Positive cells(q value&lt;0.05): red dot; Negative cells: other
dot.</p>
</div>
<div class="section level3">
<h3 id="plot-the-barplot-of-the-proportion-of-positive-cell-">4.2 Plot the barplot of the proportion of positive Cell.<a class="anchor" aria-label="anchor" href="#plot-the-barplot-of-the-proportion-of-positive-cell-"></a>
</h3>
<p>Plot the barplot of the proportion of positive Cells in
celltypes:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"D:/OneDrive/RPakage/scPagwas/vignettes/figures/plot_bar_positie_nagtive.pdf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_bar_positie_nagtive.html">plot_bar_positie_nagtive</a></span><span class="op">(</span>seurat_obj<span class="op">=</span><span class="va">Pagwas</span>,</span>
<span>                         var_ident<span class="op">=</span><span class="st">"celltypes"</span>,</span>
<span>                         var_group<span class="op">=</span><span class="st">"positiveCells"</span>,</span>
<span>                         vec_group_colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#E8D0B3"</span>,<span class="st">"#7EB5A6"</span><span class="op">)</span>,</span>
<span>                         do_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/plot_bar_positie_nagtive.png" width="60%"></p>
<p>Plot the barplot of the proportion of celltypes in positive and
negative Cells:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"D:/OneDrive/RPakage/scPagwas/vignettes/figures/plot_bar_positie_nagtive.pdf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_bar_positie_nagtive.html">plot_bar_positie_nagtive</a></span><span class="op">(</span>seurat_obj<span class="op">=</span><span class="va">Pagwas</span>,</span>
<span>                              var_ident<span class="op">=</span><span class="st">"positiveCells"</span>,</span>
<span>                              var_group<span class="op">=</span><span class="st">"celltypes"</span>,</span>
<span>                              p_thre <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                              vec_group_colors<span class="op">=</span><span class="cn">NULL</span>,</span>
<span>                              f_color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">10</span>, name<span class="op">=</span><span class="st">"RdYlBu"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                              do_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/plot_bar_positie_nagtive2.png" width="60%"></p>
</div>
<div class="section level3">
<h3 id="plot-the-heritability-correlated-genes">4.3 Plot the heritability correlated genes<a class="anchor" aria-label="anchor" href="#plot-the-heritability-correlated-genes"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/heritability_cor_scatterplot.html">heritability_cor_scatterplot</a></span><span class="op">(</span>gene_heri_cor<span class="op">=</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">GeneticExpressionIndex</span>,</span>
<span>                             topn_genes_label<span class="op">=</span><span class="fl">10</span>,</span>
<span>                             color_low<span class="op">=</span><span class="st">"#035397"</span>,</span>
<span>                             color_high <span class="op">=</span><span class="st">"#F32424"</span>,</span>
<span>                             color_mid <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                             text_size<span class="op">=</span><span class="fl">2</span>,</span>
<span>                             do_plot<span class="op">=</span><span class="cn">T</span>,</span>
<span>                             max.overlaps <span class="op">=</span><span class="fl">20</span>,</span>
<span>                             width <span class="op">=</span> <span class="fl">7</span>,</span>
<span>                             height <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/heritability_cor_scatterplot.png" width="50%"></p>
</div>
<div class="section level3">
<h3 id="celltypes-bootstrap_results-reuslt">4.5 celltypes bootstrap_results reuslt<a class="anchor" aria-label="anchor" href="#celltypes-bootstrap_results-reuslt"></a>
</h3>
<p>Barplot for celltypes</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Bootstrap_P_Barplot.html">Bootstrap_P_Barplot</a></span><span class="op">(</span>p_results<span class="op">=</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">bootstrap_results</span><span class="op">$</span><span class="va">bp_value</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                    p_names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">bootstrap_results</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                    figurenames <span class="op">=</span> <span class="st">"Bootstrap_P_Barplot.pdf"</span>,</span>
<span>                    width <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                    height <span class="op">=</span> <span class="fl">7</span>,</span>
<span>                    do_plot<span class="op">=</span><span class="cn">T</span>,</span>
<span>                    title <span class="op">=</span> <span class="st">"monocytecount_bmmc"</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/Bootstrap_P_Barplot.png" width="40%"></p>
<p>Forest plot for estimate value.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Bootstrap_estimate_Plot.html">Bootstrap_estimate_Plot</a></span><span class="op">(</span>bootstrap_results<span class="op">=</span><span class="va">Pagwas</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">bootstrap_results</span>,</span>
<span>                        figurenames <span class="op">=</span> <span class="st">"estimateplot.pdf"</span>,</span>
<span>                        width <span class="op">=</span> <span class="fl">9</span>,</span>
<span>                        height <span class="op">=</span> <span class="fl">7</span>,</span>
<span>                        do_plot<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/estimate_Plot_00.png" width="60%"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.2 (2022-10-31 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 22621)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_China.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_China.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_China.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                               </span></span>
<span><span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_China.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] knitr_1.40</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] rstudioapi_0.14   magrittr_2.0.3    R6_2.5.1          ragg_1.2.4       </span></span>
<span><span class="co">#&gt;  [5] rlang_1.0.6       fastmap_1.1.0     stringr_1.4.1     tools_4.2.2      </span></span>
<span><span class="co">#&gt;  [9] xfun_0.34         cli_3.3.0         jquerylib_0.1.4   htmltools_0.5.3  </span></span>
<span><span class="co">#&gt; [13] systemfonts_1.0.4 yaml_2.3.6        digest_0.6.30     rprojroot_2.0.3  </span></span>
<span><span class="co">#&gt; [17] pkgdown_2.0.6     textshaping_0.3.6 purrr_0.3.4       sass_0.4.2       </span></span>
<span><span class="co">#&gt; [21] fs_1.5.2          memoise_2.0.1     cachem_1.0.6      evaluate_0.18    </span></span>
<span><span class="co">#&gt; [25] rmarkdown_2.18    stringi_1.7.8     compiler_4.2.2    bslib_0.4.1      </span></span>
<span><span class="co">#&gt; [29] desc_1.4.2        jsonlite_1.8.3    renv_0.16.0</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chunyu Deng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
