<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scPagwas">
<title>Data_input_preproccess_for_scPagwas • scPagwas</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Data_input_preproccess_for_scPagwas">
<meta property="og:description" content="scPagwas">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scPagwas</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Bmmc_monocytecount_singlecell_celltype_vignette.html">Bmmc_monocytecount_singlecell_celltype_vignette</a>
    <a class="dropdown-item" href="../articles/Data_input_preproccess_for_scPagwas.html">Data_input_preproccess_for_scPagwas</a>
    <a class="dropdown-item" href="../articles/Multi_TraitFiles_for_one_SingleData_running.html">Multi_TraitFiles_for_one_SingleData_running</a>
    <a class="dropdown-item" href="../articles/Prune_GwasSummaryFIles_plink.html">Prune_GwasSummaryFIles_plink</a>
    <a class="dropdown-item" href="../articles/Running_scPagwas_steps_by_SubFunctions.html">Running_scPagwas_steps_by_SubFunctions</a>
    <a class="dropdown-item" href="../articles/Split_big_scRNAseqData_and_integrate_result.html">Split_big_scRNAseqData_and_integrate_result</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Data_input_preproccess_for_scPagwas</h1>
            
            <h4 data-toc-skip class="date">Last Updated: 08, 三月, 2023
at 15:32</h4>
      
      
      <div class="d-none name"><code>Data_input_preproccess_for_scPagwas.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="data-input">Data input<a class="anchor" aria-label="anchor" href="#data-input"></a>
</h2>
<p>There were three input resources for scPagwas: scRNA-seq
dataset(seruat format), a GWAS summary dataset(txt file) on a given
phenotype and an extensive panel of pathways or functional gene
sets(gene symbol list).</p>
<div class="section level3">
<h3 id="single-cell-data-input">1.Single cell data Input<a class="anchor" aria-label="anchor" href="#single-cell-data-input"></a>
</h3>
<div class="section level4">
<h4 id="downloading-scrna-seq-dataset">1.1.Downloading scRNA-seq dataset<a class="anchor" aria-label="anchor" href="#downloading-scrna-seq-dataset"></a>
</h4>
<p>The example scRNA-seq data for T cells of melanoma was downloaded
from the GEO <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE115978" class="external-link">GSE115978</a>
Two data were download from this website: Single cell annotations data:
<a href="https://ftp.ncbi.nlm.nih.gov/geo/series/GSE115nnn/GSE115978/suppl/GSE115978_cell.annotations.csv.gz" class="external-link">GSE115978_cell.annotations.csv.gz</a></p>
<p>Single cell count <a href="data:%5BGSE115978_counts.csv.gz" class="uri">data:[GSE115978_counts.csv.gz</a>](<a href="https://ftp.ncbi.nlm.nih.gov/geo/series/GSE115nnn/GSE115978/suppl/GSE115978_counts.csv.gz" class="external-link uri">https://ftp.ncbi.nlm.nih.gov/geo/series/GSE115nnn/GSE115978/suppl/GSE115978_counts.csv.gz</a>)</p>
</div>
<div class="section level4">
<h4 id="progressing-scrna-seq-dataset">1.2.Progressing scRNA-seq dataset<a class="anchor" aria-label="anchor" href="#progressing-scrna-seq-dataset"></a>
</h4>
<ul>
<li>1). The scRNA-seq dataset should have Idents for cell types or
clusters.</li>
<li>2). The scRNA-seq dataset should be normalized.</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="gwas-summary-data-input">2.GWAS summary data Input<a class="anchor" aria-label="anchor" href="#gwas-summary-data-input"></a>
</h3>
<p>GWAS Summary statistics are download from figshare: <a href="https://doi.org/10.6084/m9.figshare.14671995" class="external-link">pgc.bip.full.2012-04.txt.gz</a></p>
<div class="section level4">
<h4 id="read-and-progrocess-the-example-gwas-summary-statistics-file">2.1.Read and progrocess the example GWAS Summary statistics
file<a class="anchor" aria-label="anchor" href="#read-and-progrocess-the-example-gwas-summary-statistics-file"></a>
</h4>
<p>In R environment, select the specific columns and output the
result.</p>
<p>There were more than 10M snp in the GWAS summary datasets. To
facilitate the calculation of scPagwas, we prune the snp by plink. Here,
we</p>
<p><code>awk  '{print $3 }' ${i}.txt  &gt; /share/pub/dengcy/Cancer_Gwas/CollectedData/Gwas_data/IEU_opengwas_project/tempfile/${i}_SNP_list.txt</code></p>
<p>The GWAS Summary statistics file need to be processed into a “txt”
file including six coloumn and tab-delimited.</p>
</div>
<div class="section level4">
<h4 id="take-the-monocyte-count-as-real-example-and-deal-with-it-step-by-step">2.2.Take the monocyte count as real example and deal with it step by
step:<a class="anchor" aria-label="anchor" href="#take-the-monocyte-count-as-real-example-and-deal-with-it-step-by-step"></a>
</h4>
<p>Preprogress the raw monocytecount GWAS Summary statistics file.
Extract the useful information from raw files. Download the Raw gwas
GWAS Summary statistics file: <a href="https://gwas.mrcieu.ac.uk/files/ieu-b-31/ieu-b-31.vcf.gz" class="external-link uri">https://gwas.mrcieu.ac.uk/files/ieu-b-31/ieu-b-31.vcf.gz</a></p>
</div>
</div>
<div class="section level3">
<h3 id="pathway-gene-list">3.Pathway gene list<a class="anchor" aria-label="anchor" href="#pathway-gene-list"></a>
</h3>
<p>There are some processed pathway gene list provided by scPagwas
package: all these pathway list are including gene symbols.</p>
<ul>
<li>Genes_by_pathway_kegg</li>
<li>genes.by.celltype.pathway</li>
<li>genes.by.gpbp.pathway</li>
<li>genes.by.reactome.pathway</li>
<li>genes.by.regulatory.pathway</li>
<li>genes.by.tft.pathway</li>
<li>genes.by.hallmark.pathway</li>
<li>genes.by.immunologic.pathway</li>
<li>genes.by.immunesigdb.pathway</li>
</ul>
<p>There are some steps for get these pathway gene list: There is no
need to repeat running these code.</p>
<div class="section level4">
<h4 id="kegg-pathway-list">KEGG pathway list<a class="anchor" aria-label="anchor" href="#kegg-pathway-list"></a>
</h4>
</div>
<div class="section level4">
<h4 id="other-pathway-list">Other pathway list<a class="anchor" aria-label="anchor" href="#other-pathway-list"></a>
</h4>
<p>These pathways data are download from <a href="https://www.gsea-msigdb.org/gsea/index.jsp" class="external-link">GSEA</a> website.</p>
<p>Note:</p>
<ol style="list-style-type: decimal">
<li>Sometimes, there is no need to change the pathway list frequently
once you choose a fitable pathway list.</li>
<li>The summed number of genes in pathway list should not too much to
cost a long time or memory.Once your pathway list is too big, you can
choose some sub-pathway suitable for you analysisi or remove some
redundance pathways.</li>
</ol>
</div>
<div class="section level4">
<h4 id="reduce-pathway-gene-list">Reduce pathway gene list<a class="anchor" aria-label="anchor" href="#reduce-pathway-gene-list"></a>
</h4>
<p>We set the <code>genes.by.reactome.pathway</code> for example, there
are 1615 gene list and 89476 genes, a mount of resouces and time will be
cost when running scPagwas, we provid <code>reduce_pathway</code> to
reduce the pathway list:</p>
<ul>
<li>
<code>pathway_seed</code> choose a list of pathway names as seed,
which are some pahtways cann’t be remove.</li>
<li>
<code>remove_proporion</code> The propotion of duplicated between
seed pathway and the others</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="gene-block-annotation">4.Gene block annotation<a class="anchor" aria-label="anchor" href="#gene-block-annotation"></a>
</h3>
<p>Gene block annotation in chromosome is need to prepare for scPagwas.
scPagwas can provide a block annotation data for protein-coding
genes.</p>
<p>File downloaded from MAGMA website <a href="https://ctg.cncr.nl/software/magma" class="external-link">MAGMA | CTG (cncr.nl)</a></p>
<p>Here is the procedure of obtaining the data:</p>
</div>
<div class="section level3">
<h3 id="ld-data">5.LD data<a class="anchor" aria-label="anchor" href="#ld-data"></a>
</h3>
<p>The 1,000 Genomes Project Phase 3 Panel was applied to calculate the
linkage disequilibrium (LD) among SNPs extracted from GWAS summary
statistics. the processed LD data are show here:</p>
<p>We use <code>vcftools</code> and <code>plink</code> to deal with the
1,000 Genomes genotypes.</p>
<p><code>./vcftools --vcf ./1000genomes_all_genotypes.vcf --plink-tped --out ./1000genomes_all_genotypes ./plink --tfile ./1000genomes_all_genotypes --recode --out ./1000genomes_all_genotypes ./plink --map ./1000genomes_all_genotypes.map --ped ./1000genomes_all_genotypes.ped --allow-no-sex --autosome --r2 --ld-window-kb 1000 --ld-window-r2 0.2 --out ./ld_1000genome</code></p>
<p>R environment to print out the scPagwas-needed data.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chunyu Deng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
