<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scPagwas">
<title>Introduction to Data Input and Preprocessing in scPagwas • scPagwas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to Data Input and Preprocessing in scPagwas">
<meta property="og:description" content="scPagwas">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scPagwas</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Conventional%20Parameters%20and%20Usage%20Instructions%20with%20Demo%20Example%20Data.html">Conventional Parameters and Usage Instructions with Demo Example Data</a>
    <a class="dropdown-item" href="../articles/Conventional%20result%20and%20visualization%20Instructions%20with%20Real-World%20Examples.html">Conventional result and visualization Instructions with Real-World Examples</a>
    <a class="dropdown-item" href="../articles/Introduction%20to%20Data%20Input%20and%20Preprocessing%20in%20scPagwas.html">Introduction to Data Input and Preprocessing in scPagwas</a>
    <a class="dropdown-item" href="../articles/Perform%20calculations%20for%20multiple%20traits%20based%20on%20a%20single-cell%20dataset.html">Perform calculations for multiple traits based on a single-cell dataset</a>
    <a class="dropdown-item" href="../articles/Pruning%20Process%20for%20GWAS%20Summary%20Statistics%20File%20in%20scPagwas.html">Pruning Process for GWAS Summary Statistics File in scPagwas</a>
    <a class="dropdown-item" href="../articles/Strategies%20for%20Large-scale%20Single-cell%20Data%20Subsetting%20and%20Computation.html">Strategies for Large-scale Single-cell Data Subsetting and Computation</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to Data Input and Preprocessing in scPagwas</h1>
            
            <h4 data-toc-skip class="date">Last Updated: 31, 五月, 2023
at 09:28</h4>
      
      
      <div class="d-none name"><code>Introduction to Data Input and Preprocessing in scPagwas.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="data-input">Data input<a class="anchor" aria-label="anchor" href="#data-input"></a>
</h2>
<ul>
<li>Resources1: There were tow input files for you must to prepare in
scPagwas: scRNA-seq dataset(seruat format), a GWAS summary dataset(txt
file);</li>
<li>Resources2: There were three input files for you choosing to prepare
in scPagwas: an extensive panel of pathways or functional gene sets(gene
symbol list); Gene block annotation nd LD data.</li>
</ul>
<div class="section level3">
<h3 id="resources1">Resources1<a class="anchor" aria-label="anchor" href="#resources1"></a>
</h3>
<div class="section level4">
<h4 id="single-cell-data-input">1.Single cell data Input<a class="anchor" aria-label="anchor" href="#single-cell-data-input"></a>
</h4>
<div class="section level5">
<h5 id="downloading-scrna-seq-dataset">1.1.Downloading scRNA-seq dataset<a class="anchor" aria-label="anchor" href="#downloading-scrna-seq-dataset"></a>
</h5>
<p>The example scRNA-seq data for T cells of melanoma was downloaded
from the GEO <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE115978" class="external-link">GSE115978</a>
Two data were download from this website: Single cell annotations data:
<a href="https://ftp.ncbi.nlm.nih.gov/geo/series/GSE115nnn/GSE115978/suppl/GSE115978_cell.annotations.csv.gz" class="external-link">GSE115978_cell.annotations.csv.gz</a></p>
<p>Single cell count <a href="data:%5BGSE115978_counts.csv.gz" class="uri">data:[GSE115978_counts.csv.gz</a>](<a href="https://ftp.ncbi.nlm.nih.gov/geo/series/GSE115nnn/GSE115978/suppl/GSE115978_counts.csv.gz" class="external-link uri">https://ftp.ncbi.nlm.nih.gov/geo/series/GSE115nnn/GSE115978/suppl/GSE115978_counts.csv.gz</a>)</p>
</div>
<div class="section level5">
<h5 id="progressing-scrna-seq-dataset">1.2.Progressing scRNA-seq dataset<a class="anchor" aria-label="anchor" href="#progressing-scrna-seq-dataset"></a>
</h5>
<p>To analyze scRNA-seq data using scPagwas, it is essential to have an
RNA assay included. If working with raw scRNA-seq data, further
preprocessing steps such as clustering and annotation are required.
After clustering, it is necessary to assign the annotated information to
the “Idents” variable. Additionally, the gene names in scRNA-seq data
should be represented using gene symbols. Since we have downloaded
pre-annotated scRNA-seq data, we will skip the steps of clustering and
annotation in this analysis. Therefore, we won’t be performing any
clustering or annotation on the dataset. - 1). The scRNA-seq dataset
should have Idents for cell types or clusters. - 2). The scRNA-seq
dataset should be normalized.</p>
</div>
</div>
<div class="section level4">
<h4 id="gwas-summary-data-input">2.GWAS summary data Input<a class="anchor" aria-label="anchor" href="#gwas-summary-data-input"></a>
</h4>
<p>GWAS Summary statistics are download from Finngen: <a href="https://storage.googleapis.com/finngen-public-data-r7/summary_stats/finngen_R7_C3_GBM_EXALLC.gz" class="external-link">GBM_EXALLC</a></p>
<div class="section level5">
<h5 id="read-and-progrocess-the-example-gwas-summary-statistics-file">2.1.Read and progrocess the example GWAS Summary statistics
file<a class="anchor" aria-label="anchor" href="#read-and-progrocess-the-example-gwas-summary-statistics-file"></a>
</h5>
<p>In R environment, select the specific columns and output the
result.</p>
<p>The GWAS Summary statistics file need to be processed into a “txt”
file including six coloumn and tab-delimited.</p>
<p>In the context of scPagwas software, the presence of specific columns
in the GWAS summary file is generally required. However, there are
situations where the obtained GWAS summary may not perfectly meet these
requirements. In such cases, there are alternative approaches that can
be employed.</p>
<p>Among the six columns typically present in the GWAS summary file, the
“maf” column is primarily used in scPagwas for SNP filtering purposes.
It allows for the reduction of SNP quantity and computational burden.
However, it does not play a critical role in the analysis. As a
workaround, it is possible to customize this column to include any value
above the minimum minor allele frequency (maf), essentially bypassing
the maf filtering step. This modification has no significant impact on
the results since scPagwas aims to integrate a larger number of SNP
data, regardless of the magnitude of their effects.</p>
<p>In GWAS summary data, the effect size (ES) and standard error (SE)
for each SNP locus are typically provided. In cases where beta data is
unavailable, it is possible to compute a conversion using the following
formula: beta = ES / SE Here, beta represents the effect size of the SNP
locus. Alternatively, the ES column itself can be used to represent the
impact of the SNP locus on the target trait. Therefore, replacing the
beta column with the ES values directly for scPagwas calculations is
also acceptable. However, it is important to maintain consistent column
names as per the example data.</p>
<p>Additionally, both TXT files and data frames are acceptable formats
for input data.</p>
<p>Please note that scPagwas analysis supports these adaptations to
accommodate GWAS summary files that do not perfectly align with the
standard requirements.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="resource2">Resource2<a class="anchor" aria-label="anchor" href="#resource2"></a>
</h3>
<p>There were three input resources in scPagwas (Most of time need not
prepare): an extensive panel of pathways or functional gene sets(gene
symbol list); and LD data.</p>
<div class="section level4">
<h4 id="pathway-gene-list">3.Pathway gene list<a class="anchor" aria-label="anchor" href="#pathway-gene-list"></a>
</h4>
<p>There are some processed pathway gene list provided by scPagwas
package: all these pathway list are including gene ids; Gene block
annotation(the same id with pathway list);</p>
<ul>
<li>Genes_by_pathway_kegg</li>
<li>genes.by.gpbp.pathway</li>
<li>genes.by.reactome.pathway</li>
<li>genes.by.regulatory.pathway</li>
<li>genes.by.tft.pathway</li>
<li>reduce_genes.by.gpbp.pathway</li>
<li>reduce_genes.by.reactome.pathway</li>
<li>reduce_genes.by.regulatory.pathway</li>
<li>reduce_genes.by.tft.pathway</li>
</ul>
<p>When selecting pathway data, there are several principles to consider
based on the scPagwas software package. Firstly, it is recommended to
prioritize pathways with higher universality and broader coverage, such
as Genes_by_pathway_kegg and genes.by.gpbp.pathway. Single functional
pathways like hallmark and immunologic have not been thoroughly tested
for effectiveness. Secondly, the number of pathways included in the list
affects both the result and time efficiency of the analysis. Too many
pathways can significantly increase computation time, while too few may
not provide sufficient coverage. Typically, a reasonable range is
between 200 to 1000 pathways, with a gene coverage of at least 50% when
compared to the genes in the single-cell expression profile data. It is
advisable to use established pathways like Genes_by_pathway_kegg and
genes.by.gpbp.pathway (or reduced_genes.by.gpbp.pathway with appropriate
redundancy removal) to ensure result efficacy. The effectiveness of
these pathways has been extensively validated in published scPagwas
articles.</p>
<p>Examples of pathway acquisition and redundancy removal are provided
below for reference. There is no need to repeat running these code.</p>
<div class="section level5">
<h5 id="kegg-pathway-list">KEGG pathway list<a class="anchor" aria-label="anchor" href="#kegg-pathway-list"></a>
</h5>
</div>
<div class="section level5">
<h5 id="other-pathway-list">Other pathway list<a class="anchor" aria-label="anchor" href="#other-pathway-list"></a>
</h5>
<p>These pathways data are download from <a href="https://www.gsea-msigdb.org/gsea/index.jsp" class="external-link">GSEA</a> website.</p>
<p>Note:</p>
<ol style="list-style-type: decimal">
<li>Sometimes, there is no need to change the pathway list frequently
once you choose a fitable pathway list.</li>
<li>The summed number of genes in pathway list should not too much to
cost a long time or memory.Once your pathway list is too big, you can
choose some sub-pathway suitable for you analysis or remove some
redundance pathways.</li>
</ol>
</div>
<div class="section level5">
<h5 id="reduce-pathway-gene-list">Reduce pathway gene list<a class="anchor" aria-label="anchor" href="#reduce-pathway-gene-list"></a>
</h5>
<p>We set the <code>genes.by.reactome.pathway</code> for example, there
are 1615 gene list and 89476 genes, a mount of resouces and time will be
cost when running scPagwas, we provid <code>reduce_pathway</code> to
reduce the pathway list:</p>
<ul>
<li>
<code>pathway_seed</code> choose a list of pathway names as seed,
which are the pahtways cann’t be remove.</li>
<li>
<code>remove_proporion</code> The propotion of duplicated between
seed pathway and the others</li>
</ul>
<blockquote>
<p>Note.The summed genes for all pathways should not smaller than 1/2 of
the number of genes in single cell data.</p>
</blockquote>
</div>
</div>
<div class="section level4">
<h4 id="gene-block-annotation">4.Gene block annotation<a class="anchor" aria-label="anchor" href="#gene-block-annotation"></a>
</h4>
<p>Gene block annotation in chromosome is need to prepare for scPagwas.
scPagwas can provide a block annotation data for protein-coding
genes.</p>
<p>File downloaded from MAGMA website <a href="https://ctg.cncr.nl/software/magma" class="external-link">MAGMA | CTG (cncr.nl)</a></p>
<p>Here is the procedure of obtaining the data:</p>
<p>The block_annotation data provided by scPagwas is used to obtain the
precise coordinates of genes, which enables the determination of the TSS
range of each gene based on the size of the window. However, this data
is not optimized for long-range enhancer regulation, which is why other
methods like sclinker attempt to incorporate this information into the
analysis. While block_annotation can be customized to suit specific
annotation data formats, we consider TSS regulation to be more reliable
and therefore prefer this approach.</p>
</div>
<div class="section level4">
<h4 id="ld-data">5.LD data<a class="anchor" aria-label="anchor" href="#ld-data"></a>
</h4>
<p>The LD data provided by the scPagwas software package is fixed and
generally does not require modification. The 1,000 Genomes Project Phase
3 Panel was applied to calculate the linkage disequilibrium (LD) among
SNPs extracted from GWAS summary statistics. the processed LD data are
show here:</p>
<p>We use <code>vcftools</code> and <code>plink</code> to deal with the
1,000 Genomes genotypes.</p>
<p><code>./vcftools --vcf ./1000genomes_all_genotypes.vcf --plink-tped --out ./1000genomes_all_genotypes ./plink --tfile ./1000genomes_all_genotypes --recode --out ./1000genomes_all_genotypes ./plink --map ./1000genomes_all_genotypes.map --ped ./1000genomes_all_genotypes.ped --allow-no-sex --autosome --r2 --ld-window-kb 1000 --ld-window-r2 0.2 --out ./ld_1000genome</code></p>
<p>R environment to print out the scPagwas-needed data.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chunyu Deng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
