<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scPagwas">
<title>Pruning Process for GWAS Summary Statistics File in scPagwas • scPagwas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pruning Process for GWAS Summary Statistics File in scPagwas">
<meta property="og:description" content="scPagwas">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scPagwas</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Conventional%20Parameters%20and%20Usage%20Instructions%20with%20Demo%20Example%20Data.html">Conventional Parameters and Usage Instructions with Demo Example Data</a>
    <a class="dropdown-item" href="../articles/Conventional%20result%20and%20visualization%20Instructions%20with%20Real-World%20Examples.html">Conventional result and visualization Instructions with Real-World Examples</a>
    <a class="dropdown-item" href="../articles/Introduction%20to%20Data%20Input%20and%20Preprocessing%20in%20scPagwas.html">Introduction to Data Input and Preprocessing in scPagwas</a>
    <a class="dropdown-item" href="../articles/Perform%20calculations%20for%20multiple%20traits%20based%20on%20a%20single-cell%20dataset.html">Perform calculations for multiple traits based on a single-cell dataset</a>
    <a class="dropdown-item" href="../articles/Pruning%20Process%20for%20GWAS%20Summary%20Statistics%20File%20in%20scPagwas.html">Pruning Process for GWAS Summary Statistics File in scPagwas</a>
    <a class="dropdown-item" href="../articles/Strategies%20for%20Large-scale%20Single-cell%20Data%20Subsetting%20and%20Computation.html">Strategies for Large-scale Single-cell Data Subsetting and Computation</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Pruning Process for GWAS Summary Statistics File in scPagwas</h1>
            
      
      
      <div class="d-none name"><code>Pruning Process for GWAS Summary Statistics File in scPagwas.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scPagwas</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="pruning-process-for-gwas-summary-statistics-file-in-scpagwas">Pruning Process for GWAS Summary Statistics File in scPagwas<a class="anchor" aria-label="anchor" href="#pruning-process-for-gwas-summary-statistics-file-in-scpagwas"></a>
</h2>
<p>The GWAS summary statistics file needs to be provided by users.
However, in many cases, the number of SNPs in such files can reach ten
millions, and this can significantly impact computing time. Therefore,
it is recommended to preprocess the file using Plink software’s prune
function before inputting it. The prune function is a commonly used tool
in Plink software that selects representative SNPs between LD loci,
reduces the number of highly similar loci, and lowers the false-positive
rate, thereby improving result reliability. Furthermore, this process
does not significantly impact the final result. Hence, it is advisable
to use this method if the GWAS file is too large.</p>
<div class="section level3">
<h3 id="progressing-gwas-data">Progressing gwas data<a class="anchor" aria-label="anchor" href="#progressing-gwas-data"></a>
</h3>
<p>To calculate plink, important information from the GWAS data needs to
be extracted.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">GWAS_raw</span> <span class="op">&lt;-</span><span class="fu">read_table</span><span class="op">(</span><span class="st">"./monocytecount.gz"</span><span class="op">)</span></span>
<span>  <span class="va">GWAS_raw</span><span class="op">&lt;-</span><span class="va">GWAS_raw</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">9</span>,<span class="fl">10</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">GWAS_raw</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chrom"</span>,<span class="st">"pos"</span>,<span class="st">"REF"</span>,<span class="st">"ALT"</span>,<span class="st">"rsid"</span>,<span class="st">"nearest_genes"</span>,<span class="st">"beta"</span>,<span class="st">"se"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">GWAS_raw</span>,file<span class="op">=</span><span class="st">"./monocytecount.txt"</span>,row.names<span class="op">=</span><span class="cn">F</span>,quote<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-snps">Extract SNPs<a class="anchor" aria-label="anchor" href="#extract-snps"></a>
</h3>
<pre><code>mkdir tempfile
awk  '{print $5 }' monocytecount.txt  &gt; ./tempfile/monocytecount_SNP_list.txt
</code></pre>
</div>
<div class="section level3">
<h3 id="plink">plink<a class="anchor" aria-label="anchor" href="#plink"></a>
</h3>
<pre><code>for i in $(seq 1 22)  
do 
echo $i
plink 
--bfile ./02_partitioned_LD_score_estimation/1000G_EUR_Phase3_plink/1000G.EUR.QC.$i 
--extract ./tempfile/monocytecount_SNP_list.txt 
--noweb --make-bed --out ./tempfile/1000G.EUR.QC.monocytecount_${i}_filtered
done </code></pre>
</div>
<div class="section level3">
<h3 id="filter-ld-information">filter LD information<a class="anchor" aria-label="anchor" href="#filter-ld-information"></a>
</h3>
<pre><code>for i in $(seq 1 22)  
do 
echo $i
plink 
--bfile ./tempfile/1000G.EUR.QC.monocytecount_${i}_filtered 
--indep-pairwise 50 5 0.8 
--out  ./tempfile/monocytecount_${i}_plink_prune_EUR_filtered_LD0.8
done</code></pre>
</div>
<div class="section level3">
<h3 id="integrate-the-result-files-for-1-22-chrome">Integrate the result files for 1-22 chrome<a class="anchor" aria-label="anchor" href="#integrate-the-result-files-for-1-22-chrome"></a>
</h3>
<pre><code><span><span class="va">cat</span> <span class="op">[</span><span class="va">monocytecount</span><span class="op">]</span><span class="op">*</span><span class="va">.prune.in</span> <span class="op">&gt;</span> <span class="va">monocytecount_EUR_LD0.8.prune</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="intersect-the-snp-for-all-gwas-file">Intersect the snp for all gwas file<a class="anchor" aria-label="anchor" href="#intersect-the-snp-for-all-gwas-file"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="va">gwas</span><span class="op">&lt;-</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html" class="external-link">read_table</a></span><span class="op">(</span><span class="st">"./monocytecount.txt"</span><span class="op">)</span></span>
<span><span class="va">SNP_prune</span><span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html" class="external-link">read_table</a></span><span class="op">(</span><span class="st">"./tempfile/AD_EUR_LD0.8.prune"</span><span class="op">)</span></span>
<span><span class="va">SNP_prune</span><span class="op">&lt;-</span><span class="va">SNP_prune</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">SNP_prune</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SNP_prune</span><span class="op">)</span><span class="op">&lt;-</span><span class="st">"rsid"</span></span>
<span><span class="co">#### Left Join using inner_join function </span></span>
<span><span class="va">gwas</span><span class="op">=</span> <span class="va">gwas</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">SNP_prune</span>,by<span class="op">=</span><span class="st">"rsid"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gwas</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">gwas</span>,file<span class="op">=</span><span class="st">"./monocytecount_prune_gwas_data.txt"</span>,row.names<span class="op">=</span><span class="cn">F</span>,quote<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chunyu Deng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
