<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='A few numbers from 
library(KEGGREST)
pathways.list &amp;lt;- keggList("pathway", "hsa")
# Pull all genes for each pathway
pathway.codes &amp;lt;- sub("path:", "", names(pathways.list))
Genes_by_pathway.kegg &amp;lt;- sapply(pathway.codes,function(pwid){
pw &amp;lt;- keggGet(pwid)
if (is.null(pw[[1]]$GENE)) return(NA)
pw2 &amp;lt;- pw[[1]]$GENE[c(FALSE,TRUE)]
# may need to modify this to c(FALSE, TRUE) for other organisms
pw2 &amp;lt;- unlist(lapply(strsplit(pw2, split = ";", fixed = T), function(x)x[1]))
 return(pw2)
})'><title>Genes_by_pathway_kegg — Genes_by_pathway_kegg • scPagwas</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genes_by_pathway_kegg — Genes_by_pathway_kegg"><meta property="og:description" content='A few numbers from 
library(KEGGREST)
pathways.list &amp;lt;- keggList("pathway", "hsa")
# Pull all genes for each pathway
pathway.codes &amp;lt;- sub("path:", "", names(pathways.list))
Genes_by_pathway.kegg &amp;lt;- sapply(pathway.codes,function(pwid){
pw &amp;lt;- keggGet(pwid)
if (is.null(pw[[1]]$GENE)) return(NA)
pw2 &amp;lt;- pw[[1]]$GENE[c(FALSE,TRUE)]
# may need to modify this to c(FALSE, TRUE) for other organisms
pw2 &amp;lt;- unlist(lapply(strsplit(pw2, split = ";", fixed = T), function(x)x[1]))
 return(pw2)
})'><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scPagwas</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Conventional%20Parameters%20and%20Usage%20Instructions%20with%20Demo%20Example%20Data.html">Conventional Parameters and Usage Instructions with Demo Example Data</a>
    <a class="dropdown-item" href="../articles/Conventional%20result%20and%20visualization%20Instructions%20with%20Real-World%20Examples.html">Conventional result and visualization Instructions with Real-World Examples</a>
    <a class="dropdown-item" href="../articles/Introduction%20to%20Data%20Input%20and%20Preprocessing%20in%20scPagwas.html">Introduction to Data Input and Preprocessing in scPagwas</a>
    <a class="dropdown-item" href="../articles/Perform%20calculations%20for%20multiple%20traits%20based%20on%20a%20single-cell%20dataset.html">Perform calculations for multiple traits based on a single-cell dataset</a>
    <a class="dropdown-item" href="../articles/Pruning%20Process%20for%20GWAS%20Summary%20Statistics%20File%20in%20scPagwas.html">Pruning Process for GWAS Summary Statistics File in scPagwas</a>
    <a class="dropdown-item" href="../articles/Strategies%20for%20Large-scale%20Single-cell%20Data%20Subsetting%20and%20Computation.html">Strategies for Large-scale Single-cell Data Subsetting and Computation</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Genes_by_pathway_kegg</h1>
      
      <div class="d-none name"><code>Genes_by_pathway_kegg.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A few numbers from <code><a href="https://rdrr.io/r/base/library.html" class="external-link">
library(KEGGREST)
pathways.list &lt;- keggList("pathway", "hsa")
# Pull all genes for each pathway
pathway.codes &lt;- sub("path:", "", names(pathways.list))
Genes_by_pathway.kegg &lt;- sapply(pathway.codes,function(pwid){
pw &lt;- keggGet(pwid)
if (is.null(pw[[1]]$GENE)) return(NA)
pw2 &lt;- pw[[1]]$GENE[c(FALSE,TRUE)]
# may need to modify this to c(FALSE, TRUE) for other organisms
pw2 &lt;- unlist(lapply(strsplit(pw2, split = ";", fixed = T), function(x)x[1]))
 return(pw2)
})</a></code></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Genes_by_pathway_kegg</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>a list.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Genes_by_pathway_kegg</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">Genes_by_pathway_kegg</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> List of 345</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00010: chr [1:67] "HK3" "HK1" "HK2" "HKDC1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00020: chr [1:30] "CS" "ACLY" "ACO2" "ACO1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00030: chr [1:30] "GPI" "G6PD" "PGLS" "H6PD" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00040: chr [1:34] "GUSB" "KL" "UGT2A1" "UGT2A3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00051: chr [1:33] "MPI" "PMM2" "PMM1" "GMPPB" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00052: chr [1:31] "GALM" "GALK1" "GALT" "GALE" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00053: chr [1:30] "UGDH" "UGT2A1" "UGT2A3" "UGT2B17" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00061: chr [1:18] "ACACA" "ACACB" "ACSF3" "MCAT" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00062: chr [1:27] "ACAA2" "HADHB" "HADH" "HADHA" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00071: chr [1:43] "ACAT2" "ACAT1" "ACAA1" "ACAA2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00072: chr [1:10] "HMGCS1" "HMGCS2" "HMGCL" "HMGCLL1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00100: chr [1:20] "FDFT1" "SQLE" "LSS" "CYP51A1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00120: chr [1:17] "CYP46A1" "CYP39A1" "HSD3B7" "CH25H" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00130: chr [1:11] "TAT" "COQ2" "COQ6" "COQ3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00140: chr [1:61] "CYP11A1" "CYP17A1" "STS" "SULT2B1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00190: chr [1:134] "ND1" "ND2" "ND3" "ND4" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00220: chr [1:22] "OTC" "ASS1" "ASL" "ARG2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00230: chr [1:130] "NUDT9" "ADPRM" "NUDT5" "PGM1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00232: chr [1:6] "CYP1A2" "NAT2" "NAT1" "CYP2A6" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00240: chr [1:56] "CAD" "DHODH" "UMPS" "CMPK1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00250: chr [1:37] "GOT1" "GOT1L1" "GOT2" "IL4I1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00260: chr [1:40] "SHMT2" "SHMT1" "AGXT" "GRHPR" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00270: chr [1:50] "CTH" "KYAT3" "KYAT1" "CBS" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00280: chr [1:48] "BCAT2" "BCAT1" "IL4I1" "BCKDHA" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00290: chr [1:4] "SDS" "SDSL" "BCAT2" "BCAT1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00310: chr [1:63] "AASS" "ALDH7A1" "AADAT" "DHTKD1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00330: chr [1:51] "GATM" "GAMT" "CKM" "CKMT1A" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00340: chr [1:22] "HAL" "UROC1" "AMDHD1" "FTCD" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00350: chr [1:36] "GOT1" "GOT1L1" "GOT2" "TAT" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00360: chr [1:17] "PAH" "DDC" "AOC3" "AOC2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00380: chr [1:42] "TDO2" "IDO1" "IDO2" "AFMID" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00400: chr [1:6] "GOT1" "GOT1L1" "GOT2" "TAT" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00410: chr [1:31] "GADL1" "CSAD" "GAD1" "GAD2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00430: chr [1:11] "CDO1" "GAD1" "GAD2" "GADL1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00440: chr [1:6] "PCYT2" "SELENOI" "CEPT1" "PCYT1B" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00450: chr [1:17] "KYAT3" "KYAT1" "MTR" "CTH" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00471: chr [1:5] "GLS2" "GLS" "GLUD2" "GLUD1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00472: chr "DAO"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00480: chr [1:58] "GGT7" "GGT6" "GGT1" "GGT5" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00500: chr [1:36] "MGAM" "MGAM2" "GAA" "GANC" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00510: chr [1:50] "SRD5A3" "DOLK" "DPAGT1" "ALG5" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00511: chr [1:18] "NEU1" "NEU3" "NEU4" "NEU2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00512: chr [1:36] "GALNTL6" "GALNT5" "GALNT17" "GALNT11" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00513: chr [1:39] "ALG13" "ALG14" "ALG1" "ALG2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00514: chr [1:47] "OGT" "EOGT" "POFUT1" "POFUT2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00515: chr [1:23] "POMT1" "POMT2" "POMGNT1" "B4GALT1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00520: chr [1:48] "CHIA" "CHIT1" "HEXA" "HEXB" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00524: chr [1:5] "HK3" "HK1" "HK2" "HKDC1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00531: chr [1:19] "HYAL2" "HYAL1" "SPAM1" "HYAL4" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00532: chr [1:20] "XYLT1" "XYLT2" "B4GALT7" "B3GALT6" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00533: chr [1:14] "FUT8" "B4GALT1" "B4GALT2" "B4GALT3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00534: chr [1:24] "XYLT1" "XYLT2" "B4GALT7" "B3GALT6" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00561: chr [1:61] "GLYCTK" "ALDH2" "ALDH3A2" "ALDH1B1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00562: chr [1:73] "PIK3C3" "MTM1" "MTMR1" "MTMR2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00563: chr [1:26] "PIGA" "PIGC" "PIGH" "PIGP" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00564: chr [1:98] "GPD1L" "GPD1" "GPD2" "GPAM" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00565: chr [1:49] "AGPS" "PLPP1" "PLPP3" "PLPP2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00590: chr [1:61] "PLA2G10" "PLA2G2D" "PLA2G2E" "PLA2G3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00591: chr [1:29] "PLA2G10" "PLA2G2D" "PLA2G2E" "PLA2G3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00592: chr [1:25] "PLA2G10" "PLA2G2D" "PLA2G2E" "PLA2G3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00600: chr [1:49] "SPTLC1" "SPTLC2" "SPTLC3" "KDSR" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00601: chr [1:27] "B3GNT5" "B3GALT1" "B3GALT2" "B3GALT5" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00603: chr [1:15] "A3GALT2" "B3GALNT1" "GBGT1" "A4GALT" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00604: chr [1:15] "B4GALNT1" "B3GALT4" "ST3GAL2" "ST3GAL1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00620: chr [1:47] "ACSS1" "ACSS2" "PDHA2" "PDHA1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00630: chr [1:30] "ACSS1" "ACSS2" "MDH1" "MDH2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00640: chr [1:34] "ACSS1" "ACSS2" "ACSS3" "BCKDHA" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00650: chr [1:28] "ACAT2" "ACAT1" "HADH" "HADHA" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00670: chr [1:20] "DHFR" "DHFR2" "MTHFD1L" "MTHFD1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00730: chr [1:16] "NFS1" "NTPCR" "ALPL" "ALPI" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00740: chr [1:8] "RFK" "ACP5" "ACP1" "ACP2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00750: chr [1:6] "PNPO" "PDXK" "PDXP" "PHOSPHO2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00760: chr [1:35] "ASPDH" "QPRT" "NAPRT" "PNP" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00770: chr [1:21] "PANK1" "PANK3" "PANK2" "PPCS" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00780: chr [1:3] "OXSM" "HLCS" "BTD"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00785: chr [1:4] "LIPT2" "LIAS" "LIPT1" "ACSM1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00790: chr [1:26] "GCH1" "ALPL" "ALPI" "ALPP" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00830: chr [1:68] "BCO1" "ADH1A" "ADH1B" "ADH1C" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00860: chr [1:43] "ALAS1" "ALAS2" "EARS2" "EPRS1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00900: chr [1:22] "ACAT2" "ACAT1" "HMGCS1" "HMGCS2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00910: chr [1:17] "GLUD2" "GLUD1" "GLUL" "CPS1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00920: chr [1:10] "PAPSS2" "PAPSS1" "BPNT1" "BPNT2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00970: chr [1:66] "TRNA" "TRNR" "TRNN" "TRND" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00980: chr [1:78] "CYP1A1" "CYP2C9" "CYP3A4" "CYP1B1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00982: chr [1:72] "CYP2D6" "cytochrome P450 2D6-like [KO:K17712] [EC:1.14.14.1]" "CYP2D7" "CYP2D6" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa00983: chr [1:80] "HPRT1" "IMPDH1" "IMPDH2" "GMPS" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01040: chr [1:27] "ELOVL1" "ELOVL2" "ELOVL3" "ELOVL4" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01100: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01200: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01210: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01212: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01230: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01240: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01521: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01522: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01523: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa01524: logi NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa02010: chr [1:45] "DEFB1" "ABCA1" "ABCA2" "ABCA3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ hsa03008: chr [1:110] "CSNK2B" "CSNK2A1" "CSNK2A2" "CSNK2A3" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [list output truncated]</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Chunyu Deng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

